Index: firefox-trunk-22.0~a1~hg20130306r123930/toolkit/components/search/tests/xpcshell/head_search.js
===================================================================
--- firefox-trunk-22.0~a1~hg20130306r123930.orig/toolkit/components/search/tests/xpcshell/head_search.js	2013-03-07 22:46:39.736528839 +0000
+++ firefox-trunk-22.0~a1~hg20130306r123930/toolkit/components/search/tests/xpcshell/head_search.js	2013-03-07 22:56:44.604518091 +0000
@@ -19,6 +19,44 @@
 // Need to create and register a profile folder.
 var gProfD = do_get_profile();
 
+var provider = {
+  getFile: function() {
+    throw Components.results.NS_ERROR_FAILURE;
+  },
+
+  getFiles: function(prop) {
+    if (prop == "XREExtDL") {
+      return {
+        getNext: function() {
+          throw Components.results.NS_ERROR_FAILURE;
+        },
+        hasMoreElements: function() {
+          return false;
+        },
+        QueryInterface: function(iid) {
+          if (iid.equals(Ci.nsISimpleEnumerator) ||
+              iid.equals(Ci.nsISupports)) {
+            return this;
+          }
+          throw Components.results.NS_ERROR_NO_INTERFACE;
+        }
+      };
+    }
+    throw Components.results.NS_ERROR_FAILURE;
+  },
+
+  QueryInterface: function(iid) {
+    if (iid.equals(Ci.nsIDirectoryServiceProvider2) ||
+        iid.equals(Ci.nsIDirectoryServiceProvider) ||
+        iid.equals(Ci.nsISupports)) {
+      return this;
+    }
+    throw Components.results.NS_ERROR_NO_INTERFACE;
+  }
+};
+
+Services.dirsvc.registerProvider(provider);
+
 function dumpn(text)
 {
   dump("search test: " + text + "\n");
