Index: firefox-trunk-22.0~a1~hg20130301r123350/toolkit/components/search/tests/xpcshell/test_nodb_pluschanges.js
===================================================================
--- firefox-trunk-22.0~a1~hg20130301r123350.orig/toolkit/components/search/tests/xpcshell/test_nodb_pluschanges.js	2013-03-01 12:18:24.265812107 +0000
+++ firefox-trunk-22.0~a1~hg20130301r123350/toolkit/components/search/tests/xpcshell/test_nodb_pluschanges.js	2013-03-01 12:18:24.257812107 +0000
@@ -20,9 +20,6 @@
  */
 
 const Cc = Components.classes;
-const Ci = Components.interfaces;
-const Cu = Components.utils;
-const Cr = Components.results;
 
 Cu.import("resource://testing-common/httpd.js");
 
Index: firefox-trunk-22.0~a1~hg20130301r123350/toolkit/components/search/tests/xpcshell/test_purpose.js
===================================================================
--- firefox-trunk-22.0~a1~hg20130301r123350.orig/toolkit/components/search/tests/xpcshell/test_purpose.js	2013-03-01 15:17:09.000000000 +0000
+++ firefox-trunk-22.0~a1~hg20130301r123350/toolkit/components/search/tests/xpcshell/test_purpose.js	2013-03-01 15:17:28.929427801 +0000
@@ -8,8 +8,6 @@
 
 "use strict";
 
-const Ci = Components.interfaces;
-
 Components.utils.import("resource://testing-common/httpd.js");
 
 function search_observer(aSubject, aTopic, aData) {
