Index: firefox-trunk-19.0~a1~hg20121114r113129/toolkit/mozapps/extensions/PluginProvider.jsm
===================================================================
--- firefox-trunk-19.0~a1~hg20121114r113129.orig/toolkit/mozapps/extensions/PluginProvider.jsm	2012-11-14 00:37:53.000000000 +0000
+++ firefox-trunk-19.0~a1~hg20121114r113129/toolkit/mozapps/extensions/PluginProvider.jsm	2012-11-14 16:04:56.214426093 +0000
@@ -286,17 +286,17 @@
   this.__defineGetter__("scope", function() {
     let path = aTags[0].fullpath;
     // Plugins inside the application directory are in the application scope
-    let dir = Services.dirsvc.get("APlugns", Ci.nsIFile);
+    let dir = Services.dirsvc.get("APlugns", Ci.nsIFile).normalize();
     if (path.substring(0, dir.path.length) == dir.path)
       return AddonManager.SCOPE_APPLICATION;
 
     // Plugins inside the profile directory are in the profile scope
-    dir = Services.dirsvc.get("ProfD", Ci.nsIFile);
+    dir = Services.dirsvc.get("ProfD", Ci.nsIFile).normalize();
     if (path.substring(0, dir.path.length) == dir.path)
       return AddonManager.SCOPE_PROFILE;
 
     // Plugins anywhere else in the user's home are in the user scope
-    dir = Services.dirsvc.get("Home", Ci.nsIFile);
+    dir = Services.dirsvc.get("Home", Ci.nsIFile).normalize();
     if (path.substring(0, dir.path.length) == dir.path)
       return AddonManager.SCOPE_USER;
 
