diff --git a/config/config.mk b/config/config.mk
--- a/config/config.mk
+++ b/config/config.mk
@@ -159,20 +159,26 @@ endif
 
 CC := $(CC_WRAPPER) $(CC)
 CXX := $(CXX_WRAPPER) $(CXX)
 MKDIR ?= mkdir
 SLEEP ?= sleep
 TOUCH ?= touch
 
 ifndef .PYMAKE
-PYTHONPATH = $(PYTHON) $(topsrcdir)/config/pythonpath.py
+PYPATH = $(PYTHON) $(topsrcdir)/config/pythonpath.py
 else
 PYCOMMANDPATH += $(topsrcdir)/config
-PYTHONPATH = %pythonpath main
+PYPATH = %pythonpath main
+endif
+
+ifndef PYTHONPATH
+PYTHONPATH := $(PYPATH)
+else
+PYTHONPATH := $(PYPATH):$(PYTHONPATH)
 endif
 
 # determine debug-related options
 _DEBUG_CFLAGS :=
 _DEBUG_LDFLAGS :=
 
 ifdef MOZ_DEBUG
   _DEBUG_CFLAGS += $(MOZ_DEBUG_ENABLE_DEFS) $(MOZ_DEBUG_FLAGS)
