Description: Allow use of lockPref in all preference files
Author: Alexander Sack <asac@ubuntu.com>
Bug: https://bugzilla.mozilla.org/show_bug.cgi?id=467738
Forwarded: https://bugzilla.mozilla.org/attachment.cgi?id=351145

Index: firefox-trunk-58.0~a1~hg20171008r385036/modules/libpref/prefapi.cpp
===================================================================
--- firefox-trunk-58.0~a1~hg20171008r385036.orig/modules/libpref/prefapi.cpp
+++ firefox-trunk-58.0~a1~hg20171008r385036/modules/libpref/prefapi.cpp
@@ -1084,7 +1084,8 @@
                     PrefValue aValue,
                     PrefType aType,
                     bool aIsDefault,
-                    bool aIsStickyDefault)
+                    bool aIsStickyDefault,
+                    bool aIsLocked)
 {
   uint32_t flags = 0;
   if (aIsDefault) {
@@ -1096,4 +1097,6 @@
     flags |= kPrefForceSet;
   }
   pref_HashPref(aPref, aValue, aType, flags);
+  if (aIsLocked)
+    PREF_LockPref(aPref, true);
 }
Index: firefox-trunk-58.0~a1~hg20171008r385036/modules/libpref/prefapi.h
===================================================================
--- firefox-trunk-58.0~a1~hg20171008r385036.orig/modules/libpref/prefapi.h
+++ firefox-trunk-58.0~a1~hg20171008r385036/modules/libpref/prefapi.h
@@ -246,7 +246,8 @@
                           PrefValue   value,
                           PrefType    type,
                           bool        isDefault,
-                          bool        isStickyDefault);
+                          bool        isStickyDefault,
+                          bool        isLocked);
 
 
 /*
Index: firefox-trunk-58.0~a1~hg20171008r385036/modules/libpref/prefread.cpp
===================================================================
--- firefox-trunk-58.0~a1~hg20171008r385036.orig/modules/libpref/prefread.cpp
+++ firefox-trunk-58.0~a1~hg20171008r385036/modules/libpref/prefread.cpp
@@ -47,6 +47,7 @@
 #define BITS_PER_HEX_DIGIT 4
 
 static const char kUserPref[] = "user_pref";
+static const char kLockPref[] = "lockPref";
 static const char kPref[] = "pref";
 static const char kPrefSticky[] = "sticky_pref";
 static const char kTrue[] = "true";
@@ -154,7 +155,8 @@
                  value,
                  aPS->vtype,
                  aPS->fdefault,
-                 aPS->fstickydefault);
+                 aPS->fstickydefault,
+                 aPS->flock);
   return true;
 }
 
@@ -225,6 +227,7 @@
           aPS->vtype = PrefType::Invalid;
           aPS->fdefault = false;
           aPS->fstickydefault = false;
+          aPS->flock = false;
         }
         switch (c) {
           case '/': // begin comment block or line?
@@ -235,11 +238,14 @@
             break;
           case 'u': // indicating user_pref
           case 's': // indicating sticky_pref
+          case 'l': // indicating lockPref
           case 'p': // indicating pref
             if (c == 'u') {
               aPS->smatch = kUserPref;
             } else if (c == 's') {
               aPS->smatch = kPrefSticky;
+            } else if (c == 'l') {
+              aPS->smatch = kLockPref;
             } else {
               aPS->smatch = kPref;
             }
@@ -287,8 +293,9 @@
       // name parsing
       case PREF_PARSE_UNTIL_NAME:
         if (c == '\"' || c == '\'') {
-          aPS->fdefault = (aPS->smatch == kPref || aPS->smatch == kPrefSticky);
+          aPS->fdefault = (aPS->smatch == kPref || aPS->smatch == kPrefSticky || aPS->smatch == kLockPref);
           aPS->fstickydefault = (aPS->smatch == kPrefSticky);
+          aPS->flock = (aPS->smatch == kLockPref);
           aPS->quotechar = c;
           aPS->nextstate = PREF_PARSE_UNTIL_COMMA; // return here when done
           state = PREF_PARSE_QUOTED_STRING;
Index: firefox-trunk-58.0~a1~hg20171008r385036/modules/libpref/prefread.h
===================================================================
--- firefox-trunk-58.0~a1~hg20171008r385036.orig/modules/libpref/prefread.h
+++ firefox-trunk-58.0~a1~hg20171008r385036/modules/libpref/prefread.h
@@ -34,7 +34,8 @@
                            PrefValue   val,
                            PrefType    type,
                            bool        defPref,
-                           bool        stickyPref);
+                           bool        stickyPref,
+                           bool        lockPref);
 
 /**
  * Report any errors or warnings we encounter during parsing.
@@ -62,6 +63,7 @@
     PrefType    vtype;      /* PREF_STRING,INT,BOOL          */
     bool        fdefault;   /* true if (default) pref        */
     bool        fstickydefault; /* true if (sticky) pref     */
+    bool        flock;      /* true if (locked) pref         */
 } PrefParseState;
 
 /**
