---
 browser/installer/package-manifest.in |    6 ++++++
 1 file changed, 6 insertions(+)

Index: mozilla/browser/installer/package-manifest.in
===================================================================
--- mozilla.orig/browser/installer/package-manifest.in
+++ mozilla/browser/installer/package-manifest.in
@@ -358,16 +358,22 @@
 @BINPATH@/chrome/browser.manifest
 @BINPATH@/extensions/{972ce4c6-7e08-4474-a285-3208198ce6fd}/install.rdf
 @BINPATH@/extensions/{972ce4c6-7e08-4474-a285-3208198ce6fd}/icon.png
 @BINPATH@/extensions/{972ce4c6-7e08-4474-a285-3208198ce6fd}/preview.png
 @BINPATH@/chrome/toolkit.jar
 @BINPATH@/chrome/toolkit.manifest
 @BINPATH@/@PREF_DIR@/reporter.js
 #ifdef XP_UNIX
+; ubuntu awesome browser branding
+@BINPATH@/chrome/awesome-branding.jar
+@BINPATH@/chrome/awesome-branding.manifest
+@BINPATH@/chrome/awesome-branding-en-US.jar
+@BINPATH@/chrome/awesome-branding-en-US.manifest
+
 #ifndef XP_MACOSX
 @BINPATH@/chrome/icons/default/default16.png
 @BINPATH@/chrome/icons/default/default32.png
 @BINPATH@/chrome/icons/default/default48.png
 #endif
 #endif
 
 
