---
 browser/installer/unix/packages-static |    6 ++++++
 1 file changed, 6 insertions(+)

Index: mozilla/browser/installer/unix/packages-static
===================================================================
--- mozilla.orig/browser/installer/unix/packages-static
+++ mozilla/browser/installer/unix/packages-static
@@ -272,16 +272,22 @@
 bin/chrome/comm.manifest
 bin/chrome/toolkit.jar
 bin/chrome/toolkit.manifest
 bin/chrome/icons/default/default16.png
 bin/chrome/icons/default/default32.png
 bin/chrome/icons/default/default48.png
 bin/@PREF_DIR@/reporter.js
 
+; ubuntu awesome browser branding
+bin/chrome/awesome-branding.jar
+bin/chrome/awesome-branding.manifest
+bin/chrome/awesome-branding-en-US.jar
+bin/chrome/awesome-branding-en-US.manifest
+
 ; shell icons
 bin/icons/*.xpm
 bin/icons/*.png
 
 ; [Default Preferences]
 ; All the pref files must be part of base to prevent migration bugs
 bin/@PREF_DIR@/firefox.js
 bin/@PREF_DIR@/firefox-branding.js
