diff -r ddc8a4489257 security/nss/build.sh
--- a/security/nss/build.sh	Wed Aug 09 20:05:23 2017 -0700
+++ b/security/nss/build.sh	Wed Aug 16 17:24:32 2017 +0200
@@ -58,7 +58,7 @@
 
 # try to guess sensible defaults
 arch=$(python "$cwd"/coreconf/detect_host_arch.py)
-if [ "$arch" = "x64" -o "$arch" = "aarch64" ]; then
+if [ "$arch" = "x64" -o "$arch" = "aarch64" -o "$arch" = "ppc64le" -o "$arch" = "s390x" ]; then
     build_64=1
 elif [ "$arch" = "arm" ]; then
     armhf=1
diff -r ddc8a4489257 security/nss/coreconf/config.gypi
--- a/security/nss/coreconf/config.gypi	Wed Aug 09 20:05:23 2017 -0700
+++ b/security/nss/coreconf/config.gypi	Wed Aug 16 17:24:32 2017 +0200
@@ -167,7 +167,7 @@
           },
         },
       }],
-      [ 'target_arch=="arm64" or target_arch=="aarch64"', {
+      [ 'target_arch=="arm64" or target_arch=="aarch64" or target_arch=="ppc64le" or target_arch=="s390x"', {
         'defines': [
           'NSS_USE_64',
         ],
