---
 config/nsinstall.c         |    3 +--
 nsprpub/config/nsinstall.c |    3 +--
 2 files changed, 2 insertions(+), 4 deletions(-)

Index: mozilla/config/nsinstall.c
===================================================================
--- mozilla.orig/config/nsinstall.c	2007-08-13 23:55:11.000000000 +0000
+++ mozilla/config/nsinstall.c	2007-08-13 23:55:26.000000000 +0000
@@ -106,18 +106,17 @@
 
     /* strip trailing "/." */
     l = strlen(path);
     if(l > 1 && path[l - 1] == '.' && path[l - 2] == '/')
         path[l - 2] = 0;
 
     while (*path == '/' && path[1] == '/')
 	path++;
-    while ((cp = strrchr(path, '/')) && cp[1] == '\0')
-	*cp = '\0';
+    for (cp = strrchr(path, '/'); cp != path && *(cp - 1) == '/'; cp--);
     if (cp && cp != path) {
 	*cp = '\0';
 	if ((lstat(path, &sb) < 0 || !S_ISDIR(sb.st_mode)) &&
 	    mkdirs(path, mode) < 0) {
 	    return -1;
 	}
 	*cp = '/';
     }
Index: mozilla/nsprpub/config/nsinstall.c
===================================================================
--- mozilla.orig/nsprpub/config/nsinstall.c	2007-08-13 23:55:16.000000000 +0000
+++ mozilla/nsprpub/config/nsinstall.c	2007-08-13 23:55:26.000000000 +0000
@@ -130,18 +130,17 @@
 mkdirs(char *path, mode_t mode)
 {
     char *cp;
     struct stat sb;
     int res;
     
     while (*path == '/' && path[1] == '/')
 	path++;
-    while ((cp = strrchr(path, '/')) && cp[1] == '\0')
-	*cp = '\0';
+    for (cp = strrchr(path, '/'); cp != path && *(cp - 1) == '/'; cp--);
     if (cp && cp != path) {
 	*cp = '\0';
 	if ((stat(path, &sb) < 0 || !S_ISDIR(sb.st_mode)) &&
 	    mkdirs(path, mode) < 0) {
 	    return -1;
 	}
 	*cp = '/';
     }
