Index: mozilla/xpinstall/packager/Packager.pm
===================================================================
--- mozilla.orig/xpinstall/packager/Packager.pm	2009-08-31 12:28:55.000000000 +0000
+++ mozilla/xpinstall/packager/Packager.pm	2009-08-31 12:31:10.000000000 +0000
@@ -253,7 +253,7 @@
     if ( $dirflag ) { # directory copy to altdest
       ($destname, $destpath, $destsuffix) = fileparse("$destpathcomp/$altdest/$File::Find::name", '\..*?$');
       # Todo: add MSDOS hack
-      $destpath =~ s|$srcdir/$line/||;  # rm info added by find
+      $destpath =~ s|\Q$srcdir\E/$line/||;  # rm info added by find
       ($debug >= 5) &&
         print " dir copy to altdest: $destpath $destname $destsuffix\n";
     } else {  # single file copy to altdest
@@ -267,7 +267,7 @@
       if ($os eq "MSDOS") {
         $destfile =~ s|\\|/|;
       }
-      $destfile =~ s|$srcdir/||;
+      $destfile =~ s|\Q$srcdir\E/||;
 
       ($destname, $destpath, $destsuffix) = fileparse("$destpathcomp/$destfile", '\..*?$');
 
