---
 browser/installer/package-manifest.in |    4 ++--
 1 file changed, 1 insertions(+), 1 deletions(-)

Index: mozilla/browser/installer/package-manifest.in
===================================================================
--- mozilla.orig/browser/installer/package-manifest.in
+++ mozilla/browser/installer/package-manifest.in
@@ -67,18 +67,18 @@
 #endif
 #endif
 
 [browser]
 ; [Base Browser Files]
 #ifndef XP_UNIX
 @BINPATH@/@MOZ_APP_NAME@.exe
 #else
 @BINPATH@/@MOZ_APP_NAME@-bin
-@BINPATH@/@MOZ_APP_NAME@
+@BINPATH@/firefox
 #endif
 @BINPATH@/application.ini
 @BINPATH@/platform.ini
 #ifdef XP_MACOSX
 @BINPATH@/plugins/DefaultPlugin.plugin/
 #elifdef XP_UNIX
 @BINPATH@/plugins/libnullplugin.so
 #elifdef XP_WIN32
