---
 browser/installer/package-manifest.in |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: mozilla/browser/installer/package-manifest.in
===================================================================
--- mozilla.orig/browser/installer/package-manifest.in
+++ mozilla/browser/installer/package-manifest.in
@@ -70,17 +70,17 @@
 #endif
 
 [browser]
 ; [Base Browser Files]
 #ifndef XP_UNIX
 @BINPATH@/@MOZ_APP_NAME@.exe
 #else
 @BINPATH@/@MOZ_APP_NAME@-bin
-@BINPATH@/@MOZ_APP_NAME@
+@BINPATH@/firefox
 #endif
 @BINPATH@/application.ini
 @BINPATH@/platform.ini
 #ifndef XP_OS2
 @BINPATH@/@DLL_PREFIX@mozsqlite3@DLL_SUFFIX@
 #else
 @BINPATH@/mozsqlt3@DLL_SUFFIX@
 #endif
