Description: Don't bundle our own hyphenation patterns
Author: Chris Coulson <chris.coulson@canonical.com>
Forwarded: not-needed
Bug-Ubuntu: https://launchpad.net/bugs/894166

Index: firefox-trunk-41.0~a1~hg20150520r244718/browser/installer/package-manifest.in
===================================================================
--- firefox-trunk-41.0~a1~hg20150520r244718.orig/browser/installer/package-manifest.in	2015-05-20 17:21:48.341912623 +0100
+++ firefox-trunk-41.0~a1~hg20150520r244718/browser/installer/package-manifest.in	2015-05-20 17:25:30.858923264 +0100
@@ -51,7 +51,7 @@
 @RESPATH@/browser/defaults/profile/localstore.rdf
 @RESPATH@/browser/defaults/profile/mimeTypes.rdf
 @RESPATH@/dictionaries/*
-@RESPATH@/hyphenation/*
+; @RESPATH@/hyphenation/*
 @RESPATH@/browser/@PREF_DIR@/firefox-l10n.js
 #ifdef HAVE_MAKENSISU
 @BINPATH@/uninstall/helper.exe
