diff -r 932388b8c22c toolkit/locales/l10n.mk
--- a/toolkit/locales/l10n.mk	Wed Aug 16 20:58:18 2017 -0700
+++ b/toolkit/locales/l10n.mk	Thu Aug 17 12:45:56 2017 +0200
@@ -167,30 +167,8 @@
 # chrome directory.
 PKG_ZIP_DIRS = chrome $(or $(DIST_SUBDIRS),$(DIST_SUBDIR))
 
-# Clone a l10n repository, either via hg or git
-# Make this a variable as it's embedded in a sh conditional
-ifeq ($(VCS_CHECKOUT_TYPE),hg)
-L10N_CO = $(HG) --cwd $(L10NBASEDIR) clone https://hg.mozilla.org/l10n-central/$(AB_CD)/
-else
-ifeq ($(VCS_CHECKOUT_TYPE),git)
-L10N_CO = $(GIT) -C $(L10NBASEDIR) clone hg://hg.mozilla.org/l10n-central/$(AB_CD)/
-else
-L10N_CO = $(error You need to use either hg or git)
-endif
-endif
-
-
 merge-%: IS_LANGUAGE_REPACK=1
-merge-%: AB_CD=$*
 merge-%:
-# For nightly builds, we automatically check out missing localizations
-# from l10n-central.
-ifdef NIGHTLY_BUILD
-	@if  ! test -d $(L10NBASEDIR)/$(AB_CD) ; then \
-		$(NSINSTALL) -D $(L10NBASEDIR) ; \
-		$(L10N_CO) ; \
-	fi
-endif
 	$(RM) -rf $(REAL_LOCALE_MERGEDIR)
 	-$(MOZILLA_DIR)/mach compare-locales $(COMPARE_LOCALES_DEFINES) --merge $(BASE_MERGE) $(srcdir)/l10n.toml $(L10NBASEDIR) $*
 
