--- a/third_party/libwebrtc/system_wrappers/system_wrappers_gn/moz.build
+++ b/third_party/libwebrtc/system_wrappers/system_wrappers_gn/moz.build
@@ -143,19 +143,10 @@
 
 if CONFIG["CPU_ARCH"] == "arm":
 
-    CXXFLAGS += [
-        "-mfpu=neon"
-    ]
-
     DEFINES["WEBRTC_ARCH_ARM"] = True
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
-    OS_LIBS += [
-        "android_support",
-        "unwind"
-    ]
-
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
