---
 browser/app/profile/firefox.js |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: mozilla/browser/app/profile/firefox.js
===================================================================
--- mozilla.orig/browser/app/profile/firefox.js
+++ mozilla/browser/app/profile/firefox.js
@@ -507,17 +507,17 @@
 //
 // The default set of web-based protocol handlers shown in the application
 // selection dialog
 pref("gecko.handlerService.schemes.webcal.0.name", "WebCal Test Handler");
 pref("gecko.handlerService.schemes.webcal.0.uriTemplate", "http://handler-test.mozilla.org/webcal?url=%s");
 
 #ifdef MOZ_SAFE_BROWSING
 // Safe browsing does nothing unless both these prefs are set.
-pref("browser.safebrowsing.enabled", true);
+pref("browser.safebrowsing.enabled", false);
 pref("browser.safebrowsing.remoteLookups", false);
 
 // Prevent loading of pages identified as malware
 pref("browser.safebrowsing.malware.enabled", true);
 
 // Non-enhanced mode (local url lists) URL list to check for updates
 pref("browser.safebrowsing.provider.0.updateURL", "http://sb.google.com/safebrowsing/downloads?client={moz:client}&appver={moz:version}&pver=2.0");
 
