Index: firefox-trunk-21.0~a1~hg20130113r118722/testing/mochitest/browser-harness.xul
===================================================================
--- firefox-trunk-21.0~a1~hg20130113r118722.orig/testing/mochitest/browser-harness.xul	2013-01-14 09:35:35.566213583 +0000
+++ firefox-trunk-21.0~a1~hg20130113r118722/testing/mochitest/browser-harness.xul	2013-01-14 09:36:02.534213104 +0000
@@ -11,6 +11,8 @@
         width="1024">
   <script type="application/javascript" src="chrome://mochikit/content/tests/SimpleTest/MozillaLogger.js"/>
   <script type="application/javascript" src="chrome://mochikit/content/tests/SimpleTest/LogController.js"/>
+  <script type="application/javascript" src="chrome://mochikit/content/tests/SimpleTest/TestRunner.js"/>
+  <script type="application/javascript" src="chrome://mochikit/content/tests/SimpleTest/setup.js"/>
   <script type="application/javascript" src="chrome://mochikit/content/chrome-harness.js"/>
   <style xmlns="http://www.w3.org/1999/xhtml"><![CDATA[
     #results {
@@ -175,10 +177,11 @@
       scriptLoader.loadSubScript('chrome://mochikit/content/server.js',
                                  srvScope);
 
-      var fileNames = [];
       var fileNameRegexp = /browser_.+\.js$/;
-      srvScope.arrayOfTestFiles(links, fileNames, fileNameRegexp);
-      return fileNames.map(function (f) new browserTest(f));
+      srvScope.arrayOfTestFiles(links, gTestList, fileNameRegexp);
+      gTestBase = "chrome://mochitests/content";
+      gTestList = filterTests(gConfig.testManifest, gConfig.runOnly);
+      return gTestList.map(function (f) new browserTest(f));
     }
 
     function setStatus(aStatusString) {
Index: firefox-trunk-21.0~a1~hg20130113r118722/testing/mochitest/tests/SimpleTest/setup.js
===================================================================
--- firefox-trunk-21.0~a1~hg20130113r118722.orig/testing/mochitest/tests/SimpleTest/setup.js	2013-01-14 09:35:35.566213583 +0000
+++ firefox-trunk-21.0~a1~hg20130113r118722/testing/mochitest/tests/SimpleTest/setup.js	2013-01-14 09:35:35.558213584 +0000
@@ -111,6 +111,7 @@
 
 
 var gTestList = [];
+var gTestBase;
 var RunSet = {}
 RunSet.runall = function(e) {
   // Filter tests to include|exclude tests based on data in params.filter.
@@ -241,10 +242,10 @@
       for (var f in runtests) {
         // Remove leading /tests/ if exists
         file = f.replace(/^\//, '')
-        file = file.replace(/^tests\//, '')
+        file = file.replace(RegExp("^" + gTestBase + "/"), '')
 
         // Match directory or filename, gTestList has tests/<path>
-        if (tmp_path.match("^tests/" + file) != null) {
+        if (tmp_path.match("^" + gTestBase + "/" + file) != null) {
           filteredTests.push(test_path);
           break;
         }
@@ -266,10 +267,10 @@
       for (var f in excludetests) {
         // Remove leading /tests/ if exists
         file = f.replace(/^\//, '')
-        file = file.replace(/^tests\//, '')
+        file = file.replace(RegExp("^" + gTestBase + "/"), '');
 
         // Match directory or filename, gTestList has tests/<path>
-        if (tmp_path.match("^tests/" + file) != null) {
+        if (tmp_path.match("^" + gTestBase + "/" + file) != null) {
           found = true;
           break;
         }
Index: firefox-trunk-21.0~a1~hg20130113r118722/testing/mochitest/harness-overlay.xul
===================================================================
--- firefox-trunk-21.0~a1~hg20130113r118722.orig/testing/mochitest/harness-overlay.xul	2013-01-14 09:35:35.566213583 +0000
+++ firefox-trunk-21.0~a1~hg20130113r118722/testing/mochitest/harness-overlay.xul	2013-01-14 09:35:35.562213584 +0000
@@ -52,6 +52,7 @@
     document.getElementById("test-table").innerHTML += tableContent;
   }
   gTestList = eval(srvScope.jsonArrayOfTestFiles(links));
+  gTestBase = "chrome://mochitests/content";
   populate();
 
   hookup();
Index: firefox-trunk-21.0~a1~hg20130113r118722/testing/mochitest/server.js
===================================================================
--- firefox-trunk-21.0~a1~hg20130113r118722.orig/testing/mochitest/server.js	2013-01-14 09:35:35.566213583 +0000
+++ firefox-trunk-21.0~a1~hg20130113r118722/testing/mochitest/server.js	2013-01-14 09:35:35.562213584 +0000
@@ -601,7 +601,7 @@
         SCRIPT({type: "text/javascript",
                  src: "/tests/SimpleTest/setup.js"}),
         SCRIPT({type: "text/javascript"},
-               "window.onload =  hookup; gTestList=" + tests + ";"
+               "window.onload =  hookup; gTestList=" + tests + "; gTestBase = \"tests\";"
         )
       ),
       BODY(
