--- a/media/webrtc/trunk/moz.build
+++ b/media/webrtc/trunk/moz.build
@@ -121,7 +121,6 @@ if CONFIG["OS_TARGET"] == "Linux":
     DIRS += [
         "/media/webrtc/trunk/webrtc/modules/desktop_capture/desktop_capture_gn",
         "/media/webrtc/trunk/webrtc/modules/desktop_capture/primitives_gn",
-        "/media/webrtc/trunk/webrtc/system_wrappers/cpu_features_linux_gn",
         "/media/webrtc/trunk/webrtc/video_engine/video_engine_gn"
     ]
 
@@ -234,6 +233,23 @@ if CONFIG["CPU_ARCH"] == "aarch64" and C
         "/media/webrtc/trunk/webrtc/modules/video_processing/video_processing_neon_gn"
     ]
 
+if CONFIG["CPU_ARCH"] == "arm" and CONFIG["OS_TARGET"] == "Linux":
+
+    DIRS += [
+        "/media/webrtc/trunk/webrtc/common_audio/common_audio_neon_c_gn",
+        "/media/webrtc/trunk/webrtc/common_audio/common_audio_neon_gn",
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/isac_neon_gn",
+        "/media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_neon_c_gn",
+        "/media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_neon_gn",
+        "/media/webrtc/trunk/webrtc/modules/video_processing/video_processing_neon_gn"
+    ]
+
+if CONFIG["CPU_ARCH"] == "ppc64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DIRS += [
+        "/media/webrtc/trunk/webrtc/system_wrappers/cpu_features_linux_gn"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "Linux":
 
     DIRS += [
--- a/media/webrtc/trunk/webrtc/common_audio/common_audio_c_gn/moz.build
+++ b/media/webrtc/trunk/webrtc/common_audio/common_audio_c_gn/moz.build
@@ -140,12 +140,6 @@ if CONFIG["OS_TARGET"] == "Linux":
     DEFINES["WEBRTC_POSIX"] = True
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
-    UNIFIED_SOURCES += [
-        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/complex_bit_reverse.c",
-        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/filter_ar_fast_q12.c",
-        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/spl_sqrt_floor.c"
-    ]
-
 if CONFIG["OS_TARGET"] == "NetBSD":
 
     DEFINES["USE_X11"] = "1"
@@ -212,6 +206,14 @@ if CONFIG["CPU_ARCH"] == "arm":
         "/media/webrtc/trunk/webrtc/common_audio/signal_processing/spl_sqrt_floor_arm.S"
     ]
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    UNIFIED_SOURCES += [
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/complex_bit_reverse.c",
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/filter_ar_fast_q12.c",
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/spl_sqrt_floor.c"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -236,6 +238,14 @@ if not CONFIG["MOZ_DEBUG"] and CONFIG["O
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    UNIFIED_SOURCES += [
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/complex_bit_reverse.c",
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/filter_ar_fast_q12.c",
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/spl_sqrt_floor.c"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -293,6 +303,12 @@ if CONFIG["CPU_ARCH"] == "aarch64" and C
     DEFINES["DISABLE_NACL"] = True
     DEFINES["NO_TCMALLOC"] = True
 
+    UNIFIED_SOURCES += [
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/complex_bit_reverse.c",
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/filter_ar_fast_q12.c",
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/spl_sqrt_floor.c"
+    ]
+
 if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "NetBSD":
 
     UNIFIED_SOURCES += [
@@ -345,6 +361,22 @@ if CONFIG["CPU_ARCH"] == "x86_64" and CO
 
     UNIFIED_SOURCES += [
         "/media/webrtc/trunk/webrtc/common_audio/signal_processing/complex_bit_reverse.c",
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/filter_ar_fast_q12.c",
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/spl_sqrt_floor.c"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    UNIFIED_SOURCES += [
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/complex_bit_reverse.c",
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/filter_ar_fast_q12.c",
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/spl_sqrt_floor.c"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    UNIFIED_SOURCES += [
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/complex_bit_reverse.c",
         "/media/webrtc/trunk/webrtc/common_audio/signal_processing/filter_ar_fast_q12.c",
         "/media/webrtc/trunk/webrtc/common_audio/signal_processing/spl_sqrt_floor.c"
     ]
--- a/media/webrtc/trunk/webrtc/common_audio/common_audio_neon_c_gn/moz.build
+++ b/media/webrtc/trunk/webrtc/common_audio/common_audio_neon_c_gn/moz.build
@@ -69,7 +69,6 @@ if CONFIG["OS_TARGET"] == "Linux":
     DEFINES["NO_TCMALLOC"] = True
     DEFINES["USE_NSS_CERTS"] = "1"
     DEFINES["USE_X11"] = "1"
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
     DEFINES["WEBRTC_LINUX"] = True
 
 if CONFIG["OS_TARGET"] == "NetBSD":
@@ -123,4 +122,8 @@ if CONFIG["CPU_ARCH"] == "aarch64" and C
 
     DEFINES["WEBRTC_ARCH_ARM64"] = True
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["WEBRTC_ARCH_ARM64"] = True
+
 Library("common_audio_neon_c_gn")
--- a/media/webrtc/trunk/webrtc/common_audio/common_audio_neon_gn/moz.build
+++ b/media/webrtc/trunk/webrtc/common_audio/common_audio_neon_gn/moz.build
@@ -68,7 +68,6 @@ if CONFIG["OS_TARGET"] == "Linux":
     DEFINES["NO_TCMALLOC"] = True
     DEFINES["USE_NSS_CERTS"] = "1"
     DEFINES["USE_X11"] = "1"
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
     DEFINES["WEBRTC_LINUX"] = True
 
 if CONFIG["OS_TARGET"] == "NetBSD":
@@ -122,4 +121,8 @@ if CONFIG["CPU_ARCH"] == "aarch64" and C
 
     DEFINES["WEBRTC_ARCH_ARM64"] = True
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["WEBRTC_ARCH_ARM64"] = True
+
 Library("common_audio_neon_gn")
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/audio_coding_gn/moz.build
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/audio_coding_gn/moz.build
@@ -116,12 +116,7 @@ if CONFIG["OS_TARGET"] == "Linux":
     DEFINES["WEBRTC_POSIX"] = True
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
-    LOCAL_INCLUDES += [
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
-    ]
-
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -191,6 +186,16 @@ if CONFIG["CPU_ARCH"] == "arm":
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/include/"
     ]
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -215,6 +220,16 @@ if not CONFIG["MOZ_DEBUG"] and CONFIG["O
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -262,6 +277,10 @@ if CONFIG["CPU_ARCH"] == "aarch64" and C
     DEFINES["DISABLE_NACL"] = True
     DEFINES["NO_TCMALLOC"] = True
 
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "NetBSD":
 
     LOCAL_INCLUDES += [
@@ -306,4 +325,24 @@ if CONFIG["CPU_ARCH"] == "x86_64" and CO
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
     ]
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("audio_coding_gn")
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/builtin_audio_decoder_factory_gn/moz.build
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/builtin_audio_decoder_factory_gn/moz.build
@@ -104,12 +104,7 @@ if CONFIG["OS_TARGET"] == "Linux":
     DEFINES["WEBRTC_POSIX"] = True
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
-    LOCAL_INCLUDES += [
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
-    ]
-
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -179,6 +174,16 @@ if CONFIG["CPU_ARCH"] == "arm":
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/include/"
     ]
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -203,6 +208,16 @@ if not CONFIG["MOZ_DEBUG"] and CONFIG["O
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -250,6 +265,10 @@ if CONFIG["CPU_ARCH"] == "aarch64" and C
     DEFINES["DISABLE_NACL"] = True
     DEFINES["NO_TCMALLOC"] = True
 
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "NetBSD":
 
     LOCAL_INCLUDES += [
@@ -294,4 +313,24 @@ if CONFIG["CPU_ARCH"] == "x86_64" and CO
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
     ]
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("builtin_audio_decoder_factory_gn")
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/isac_fix_c_gn/moz.build
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/isac_fix_c_gn/moz.build
@@ -142,14 +142,6 @@ if CONFIG["OS_TARGET"] == "Linux":
         "rt"
     ]
 
-    SOURCES += [
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/pitch_filter_c.c"
-    ]
-
-    UNIFIED_SOURCES += [
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/lattice_c.c"
-    ]
-
 if CONFIG["OS_TARGET"] == "NetBSD":
 
     DEFINES["USE_X11"] = "1"
@@ -221,6 +213,16 @@ if CONFIG["CPU_ARCH"] == "arm":
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/pitch_filter_armv6.S"
     ]
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    SOURCES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/pitch_filter_c.c"
+    ]
+
+    UNIFIED_SOURCES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/lattice_c.c"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -245,6 +247,16 @@ if not CONFIG["MOZ_DEBUG"] and CONFIG["O
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    SOURCES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/pitch_filter_c.c"
+    ]
+
+    UNIFIED_SOURCES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/lattice_c.c"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -374,6 +386,36 @@ if CONFIG["CPU_ARCH"] == "x86_64" and CO
 
     SOURCES += [
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/pitch_filter_c.c"
+    ]
+
+    UNIFIED_SOURCES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/lattice_c.c"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    SOURCES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/pitch_filter_c.c"
+    ]
+
+    UNIFIED_SOURCES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/lattice_c.c"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    SOURCES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/pitch_filter_c.c"
+    ]
+
+    UNIFIED_SOURCES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/lattice_c.c"
+    ]
+
+if CONFIG["CPU_ARCH"] == "aarch64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    SOURCES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/pitch_filter_c.c"
     ]
 
     UNIFIED_SOURCES += [
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/isac_neon_gn/moz.build
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/isac_neon_gn/moz.build
@@ -80,7 +80,6 @@ if CONFIG["OS_TARGET"] == "Linux":
     DEFINES["NO_TCMALLOC"] = True
     DEFINES["USE_NSS_CERTS"] = "1"
     DEFINES["USE_X11"] = "1"
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
     DEFINES["WEBRTC_LINUX"] = True
 
     OS_LIBS += [
@@ -138,4 +137,8 @@ if CONFIG["CPU_ARCH"] == "aarch64" and C
 
     DEFINES["WEBRTC_ARCH_ARM64"] = True
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["WEBRTC_ARCH_ARM64"] = True
+
 Library("isac_neon_gn")
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/neteq_gn/moz.build
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/neteq_gn/moz.build
@@ -151,24 +151,11 @@ if CONFIG["OS_TARGET"] == "Linux":
 
     DEFINES["USE_NSS_CERTS"] = "1"
     DEFINES["USE_X11"] = "1"
-    DEFINES["WEBRTC_CODEC_ISAC"] = True
     DEFINES["WEBRTC_LINUX"] = True
     DEFINES["WEBRTC_POSIX"] = True
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
-    LOCAL_INCLUDES += [
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
-    ]
-
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -232,6 +219,10 @@ if CONFIG["CPU_ARCH"] == "aarch64":
     DEFINES["WEBRTC_ARCH_ARM64"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/",
+    ]
+
 if CONFIG["CPU_ARCH"] == "arm":
 
     CXXFLAGS += [
@@ -252,6 +243,19 @@ if CONFIG["CPU_ARCH"] == "arm":
         "/media/webrtc/trunk/"
     ]
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    DEFINES["WEBRTC_CODEC_ISAC"] = True
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -276,6 +280,19 @@ if not CONFIG["MOZ_DEBUG"] and CONFIG["O
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["WEBRTC_CODEC_ISAC"] = True
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -368,6 +385,10 @@ if CONFIG["CPU_ARCH"] == "aarch64" and C
     DEFINES["DISABLE_NACL"] = True
     DEFINES["NO_TCMALLOC"] = True
 
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "NetBSD":
 
     DEFINES["WEBRTC_CODEC_ISAC"] = True
@@ -466,4 +487,30 @@ if CONFIG["CPU_ARCH"] == "x86_64" and CO
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
     ]
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["WEBRTC_CODEC_ISAC"] = True
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["WEBRTC_CODEC_ISAC"] = True
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("neteq_gn")
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/rent_a_codec_gn/moz.build
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/rent_a_codec_gn/moz.build
@@ -105,12 +105,7 @@ if CONFIG["OS_TARGET"] == "Linux":
     DEFINES["WEBRTC_POSIX"] = True
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
-    LOCAL_INCLUDES += [
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
-    ]
-
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -180,6 +175,16 @@ if CONFIG["CPU_ARCH"] == "arm":
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/include/"
     ]
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -204,6 +209,16 @@ if not CONFIG["MOZ_DEBUG"] and CONFIG["O
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -251,6 +266,10 @@ if CONFIG["CPU_ARCH"] == "aarch64" and C
     DEFINES["DISABLE_NACL"] = True
     DEFINES["NO_TCMALLOC"] = True
 
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "NetBSD":
 
     LOCAL_INCLUDES += [
@@ -295,4 +314,24 @@ if CONFIG["CPU_ARCH"] == "x86_64" and CO
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
     ]
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("rent_a_codec_gn")
--- a/media/webrtc/trunk/webrtc/modules/audio_device/audio_device_gn/moz.build
+++ b/media/webrtc/trunk/webrtc/modules/audio_device/audio_device_gn/moz.build
@@ -119,7 +119,6 @@ if CONFIG["OS_TARGET"] == "Linux":
     ]
 
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -187,6 +186,12 @@ if CONFIG["CPU_ARCH"] == "arm":
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -211,6 +216,12 @@ if not CONFIG["MOZ_DEBUG"] and CONFIG["O
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -232,6 +243,10 @@ if CONFIG["CPU_ARCH"] == "aarch64" and C
     DEFINES["DISABLE_NACL"] = True
     DEFINES["NO_TCMALLOC"] = True
 
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
@@ -244,4 +259,16 @@ if CONFIG["CPU_ARCH"] == "x86" and CONFI
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("audio_device_gn")
--- a/media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_c_gn/moz.build
+++ b/media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_c_gn/moz.build
@@ -216,6 +216,13 @@ if not CONFIG["MOZ_DEBUG"] and CONFIG["O
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    UNIFIED_SOURCES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_processing/ns/noise_suppression.c",
+        "/media/webrtc/trunk/webrtc/modules/audio_processing/ns/ns_core.c"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
--- a/media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_gn/moz.build
+++ b/media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_gn/moz.build
@@ -237,6 +237,10 @@ if CONFIG["CPU_ARCH"] == "ppc64":
 
     DEFINES["WEBRTC_NS_FLOAT"] = True
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    DEFINES["WEBRTC_NS_FLOAT"] = True
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -261,6 +265,10 @@ if not CONFIG["MOZ_DEBUG"] and CONFIG["O
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["WEBRTC_NS_FLOAT"] = True
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -322,4 +330,12 @@ if CONFIG["CPU_ARCH"] == "x86_64" and CO
 
     DEFINES["WEBRTC_NS_FLOAT"] = True
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["WEBRTC_NS_FLOAT"] = True
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["WEBRTC_NS_FLOAT"] = True
+
 Library("audio_processing_gn")
--- a/media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_neon_c_gn/moz.build
+++ b/media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_neon_c_gn/moz.build
@@ -71,7 +71,6 @@ if CONFIG["OS_TARGET"] == "Linux":
     DEFINES["NO_TCMALLOC"] = True
     DEFINES["USE_NSS_CERTS"] = "1"
     DEFINES["USE_X11"] = "1"
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
     DEFINES["WEBRTC_LINUX"] = True
 
     OS_LIBS += [
@@ -129,4 +128,8 @@ if CONFIG["CPU_ARCH"] == "aarch64" and C
 
     DEFINES["WEBRTC_ARCH_ARM64"] = True
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["WEBRTC_ARCH_ARM64"] = True
+
 Library("audio_processing_neon_c_gn")
--- a/media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_neon_gn/moz.build
+++ b/media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_neon_gn/moz.build
@@ -77,7 +77,6 @@ if CONFIG["OS_TARGET"] == "Linux":
     DEFINES["NO_TCMALLOC"] = True
     DEFINES["USE_NSS_CERTS"] = "1"
     DEFINES["USE_X11"] = "1"
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
     DEFINES["WEBRTC_LINUX"] = True
 
     OS_LIBS += [
@@ -135,4 +134,8 @@ if CONFIG["CPU_ARCH"] == "aarch64" and C
 
     DEFINES["WEBRTC_ARCH_ARM64"] = True
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["WEBRTC_ARCH_ARM64"] = True
+
 Library("audio_processing_neon_gn")
--- a/media/webrtc/trunk/webrtc/modules/congestion_controller/congestion_controller_gn/moz.build
+++ b/media/webrtc/trunk/webrtc/modules/congestion_controller/congestion_controller_gn/moz.build
@@ -109,7 +109,6 @@ if CONFIG["OS_TARGET"] == "Linux":
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -173,6 +172,12 @@ if CONFIG["CPU_ARCH"] == "arm":
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -197,6 +202,12 @@ if not CONFIG["MOZ_DEBUG"] and CONFIG["O
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -218,6 +229,10 @@ if CONFIG["CPU_ARCH"] == "aarch64" and C
     DEFINES["DISABLE_NACL"] = True
     DEFINES["NO_TCMALLOC"] = True
 
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
@@ -230,4 +245,16 @@ if CONFIG["CPU_ARCH"] == "x86" and CONFI
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("congestion_controller_gn")
--- a/media/webrtc/trunk/webrtc/modules/utility/utility_gn/moz.build
+++ b/media/webrtc/trunk/webrtc/modules/utility/utility_gn/moz.build
@@ -104,7 +104,6 @@ if CONFIG["OS_TARGET"] == "Linux":
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -168,6 +167,12 @@ if CONFIG["CPU_ARCH"] == "arm":
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -192,6 +197,12 @@ if not CONFIG["MOZ_DEBUG"] and CONFIG["O
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -213,6 +224,10 @@ if CONFIG["CPU_ARCH"] == "aarch64" and C
     DEFINES["DISABLE_NACL"] = True
     DEFINES["NO_TCMALLOC"] = True
 
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
@@ -225,4 +240,16 @@ if CONFIG["CPU_ARCH"] == "x86" and CONFI
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("utility_gn")
--- a/media/webrtc/trunk/webrtc/modules/video_capture/video_capture_internal_impl_gn/moz.build
+++ b/media/webrtc/trunk/webrtc/modules/video_capture/video_capture_internal_impl_gn/moz.build
@@ -125,7 +125,6 @@ if CONFIG["OS_TARGET"] == "Linux":
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -222,6 +221,12 @@ if CONFIG["CPU_ARCH"] == "arm":
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -246,6 +251,12 @@ if not CONFIG["MOZ_DEBUG"] and CONFIG["O
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -267,6 +278,10 @@ if CONFIG["CPU_ARCH"] == "aarch64" and C
     DEFINES["DISABLE_NACL"] = True
     DEFINES["NO_TCMALLOC"] = True
 
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
@@ -279,4 +294,16 @@ if CONFIG["CPU_ARCH"] == "x86" and CONFI
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("video_capture_internal_impl_gn")
--- a/media/webrtc/trunk/webrtc/modules/video_capture/video_capture_module_gn/moz.build
+++ b/media/webrtc/trunk/webrtc/modules/video_capture/video_capture_module_gn/moz.build
@@ -102,7 +102,6 @@ if CONFIG["OS_TARGET"] == "Linux":
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -166,6 +165,12 @@ if CONFIG["CPU_ARCH"] == "arm":
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -190,6 +195,12 @@ if not CONFIG["MOZ_DEBUG"] and CONFIG["O
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -211,6 +222,10 @@ if CONFIG["CPU_ARCH"] == "aarch64" and C
     DEFINES["DISABLE_NACL"] = True
     DEFINES["NO_TCMALLOC"] = True
 
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
@@ -223,4 +238,16 @@ if CONFIG["CPU_ARCH"] == "x86" and CONFI
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("video_capture_module_gn")
--- a/media/webrtc/trunk/webrtc/modules/video_coding/video_coding_gn/moz.build
+++ b/media/webrtc/trunk/webrtc/modules/video_coding/video_coding_gn/moz.build
@@ -130,7 +130,6 @@ if CONFIG["OS_TARGET"] == "Linux":
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -194,6 +193,12 @@ if CONFIG["CPU_ARCH"] == "arm":
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -218,6 +223,12 @@ if not CONFIG["MOZ_DEBUG"] and CONFIG["O
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -239,6 +250,10 @@ if CONFIG["CPU_ARCH"] == "aarch64" and C
     DEFINES["DISABLE_NACL"] = True
     DEFINES["NO_TCMALLOC"] = True
 
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
@@ -251,4 +266,16 @@ if CONFIG["CPU_ARCH"] == "x86" and CONFI
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("video_coding_gn")
--- a/media/webrtc/trunk/webrtc/modules/video_processing/video_processing_gn/moz.build
+++ b/media/webrtc/trunk/webrtc/modules/video_processing/video_processing_gn/moz.build
@@ -108,7 +108,6 @@ if CONFIG["OS_TARGET"] == "Linux":
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -172,6 +171,12 @@ if CONFIG["CPU_ARCH"] == "arm":
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -196,6 +201,12 @@ if not CONFIG["MOZ_DEBUG"] and CONFIG["O
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -217,6 +228,10 @@ if CONFIG["CPU_ARCH"] == "aarch64" and C
     DEFINES["DISABLE_NACL"] = True
     DEFINES["NO_TCMALLOC"] = True
 
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
@@ -229,4 +244,16 @@ if CONFIG["CPU_ARCH"] == "x86" and CONFI
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("video_processing_gn")
--- a/media/webrtc/trunk/webrtc/modules/video_processing/video_processing_neon_gn/moz.build
+++ b/media/webrtc/trunk/webrtc/modules/video_processing/video_processing_neon_gn/moz.build
@@ -67,7 +67,6 @@ if CONFIG["OS_TARGET"] == "Linux":
     DEFINES["NO_TCMALLOC"] = True
     DEFINES["USE_NSS_CERTS"] = "1"
     DEFINES["USE_X11"] = "1"
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
     DEFINES["WEBRTC_LINUX"] = True
 
 if CONFIG["OS_TARGET"] == "NetBSD":
@@ -121,4 +120,8 @@ if CONFIG["CPU_ARCH"] == "aarch64" and C
 
     DEFINES["WEBRTC_ARCH_ARM64"] = True
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["WEBRTC_ARCH_ARM64"] = True
+
 Library("video_processing_neon_gn")
--- a/media/webrtc/trunk/webrtc/voice_engine/audio_coder_gn/moz.build
+++ b/media/webrtc/trunk/webrtc/voice_engine/audio_coder_gn/moz.build
@@ -92,7 +92,6 @@ if CONFIG["OS_TARGET"] == "Linux":
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -154,6 +153,12 @@ if CONFIG["CPU_ARCH"] == "arm":
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -178,6 +183,12 @@ if not CONFIG["MOZ_DEBUG"] and CONFIG["O
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -199,6 +210,10 @@ if CONFIG["CPU_ARCH"] == "aarch64" and C
     DEFINES["DISABLE_NACL"] = True
     DEFINES["NO_TCMALLOC"] = True
 
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
@@ -211,4 +226,16 @@ if CONFIG["CPU_ARCH"] == "x86" and CONFI
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("audio_coder_gn")
--- a/media/webrtc/trunk/webrtc/voice_engine/file_player_gn/moz.build
+++ b/media/webrtc/trunk/webrtc/voice_engine/file_player_gn/moz.build
@@ -93,7 +93,6 @@ if CONFIG["OS_TARGET"] == "Linux":
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -155,6 +154,12 @@ if CONFIG["CPU_ARCH"] == "arm":
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -179,6 +184,12 @@ if not CONFIG["MOZ_DEBUG"] and CONFIG["O
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -200,6 +211,10 @@ if CONFIG["CPU_ARCH"] == "aarch64" and C
     DEFINES["DISABLE_NACL"] = True
     DEFINES["NO_TCMALLOC"] = True
 
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
@@ -212,4 +227,16 @@ if CONFIG["CPU_ARCH"] == "x86" and CONFI
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("file_player_gn")
--- a/media/webrtc/trunk/webrtc/voice_engine/file_recorder_gn/moz.build
+++ b/media/webrtc/trunk/webrtc/voice_engine/file_recorder_gn/moz.build
@@ -93,7 +93,6 @@ if CONFIG["OS_TARGET"] == "Linux":
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -155,6 +154,12 @@ if CONFIG["CPU_ARCH"] == "arm":
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -179,6 +184,12 @@ if not CONFIG["MOZ_DEBUG"] and CONFIG["O
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -200,6 +211,10 @@ if CONFIG["CPU_ARCH"] == "aarch64" and C
     DEFINES["DISABLE_NACL"] = True
     DEFINES["NO_TCMALLOC"] = True
 
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
@@ -212,4 +227,16 @@ if CONFIG["CPU_ARCH"] == "x86" and CONFI
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("file_recorder_gn")
