diff -r 681eb7dfa324 media/webrtc/trunk/moz.build
--- a/media/webrtc/trunk/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -121,7 +121,6 @@
     DIRS += [
         "/media/webrtc/trunk/webrtc/modules/desktop_capture/desktop_capture_gn",
         "/media/webrtc/trunk/webrtc/modules/desktop_capture/primitives_gn",
-        "/media/webrtc/trunk/webrtc/system_wrappers/cpu_features_linux_gn",
         "/media/webrtc/trunk/webrtc/video_engine/video_engine_gn"
     ]
 
@@ -223,6 +222,34 @@
         "/media/webrtc/trunk/webrtc/modules/video_processing/video_processing_sse2_gn"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DIRS += [
+        "/media/webrtc/trunk/webrtc/common_audio/common_audio_neon_c_gn",
+        "/media/webrtc/trunk/webrtc/common_audio/common_audio_neon_gn",
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/isac_neon_gn",
+        "/media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_neon_c_gn",
+        "/media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_neon_gn",
+        "/media/webrtc/trunk/webrtc/modules/video_processing/video_processing_neon_gn"
+    ]
+
+if CONFIG["CPU_ARCH"] == "arm" and CONFIG["OS_TARGET"] == "Linux":
+
+    DIRS += [
+        "/media/webrtc/trunk/webrtc/common_audio/common_audio_neon_c_gn",
+        "/media/webrtc/trunk/webrtc/common_audio/common_audio_neon_gn",
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/isac_neon_gn",
+        "/media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_neon_c_gn",
+        "/media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_neon_gn",
+        "/media/webrtc/trunk/webrtc/modules/video_processing/video_processing_neon_gn"
+    ]
+
+if CONFIG["CPU_ARCH"] == "ppc64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DIRS += [
+        "/media/webrtc/trunk/webrtc/system_wrappers/cpu_features_linux_gn"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "Linux":
 
     DIRS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/api/audio_mixer_api_gn/moz.build
--- a/media/webrtc/trunk/webrtc/api/audio_mixer_api_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/api/audio_mixer_api_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -163,4 +163,9 @@
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 Library("audio_mixer_api_gn")
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/api/call_api_gn/moz.build
--- a/media/webrtc/trunk/webrtc/api/call_api_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/api/call_api_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -163,4 +163,9 @@
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 Library("call_api_gn")
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/api/transport_api_gn/moz.build
--- a/media/webrtc/trunk/webrtc/api/transport_api_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/api/transport_api_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -159,4 +159,9 @@
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 Library("transport_api_gn")
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/api/video_frame_api_gn/moz.build
--- a/media/webrtc/trunk/webrtc/api/video_frame_api_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/api/video_frame_api_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -193,6 +193,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/audio/audio_gn/moz.build
--- a/media/webrtc/trunk/webrtc/audio/audio_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/audio/audio_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -213,6 +213,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/audio/utility/audio_frame_operations_gn/moz.build
--- a/media/webrtc/trunk/webrtc/audio/utility/audio_frame_operations_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/audio/utility/audio_frame_operations_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -183,6 +183,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/base/gtest_prod_gn/moz.build
--- a/media/webrtc/trunk/webrtc/base/gtest_prod_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/base/gtest_prod_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -159,4 +159,9 @@
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 Library("gtest_prod_gn")
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/base/rtc_base_approved_gn/moz.build
--- a/media/webrtc/trunk/webrtc/base/rtc_base_approved_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/base/rtc_base_approved_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -246,6 +246,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/base/rtc_numerics_gn/moz.build
--- a/media/webrtc/trunk/webrtc/base/rtc_numerics_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/base/rtc_numerics_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -183,6 +183,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/base/rtc_task_queue_gn/moz.build
--- a/media/webrtc/trunk/webrtc/base/rtc_task_queue_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/base/rtc_task_queue_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -259,6 +259,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/call/call_gn/moz.build
--- a/media/webrtc/trunk/webrtc/call/call_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/call/call_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -204,6 +204,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/call/call_interfaces_gn/moz.build
--- a/media/webrtc/trunk/webrtc/call/call_interfaces_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/call/call_interfaces_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -179,6 +179,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/common_audio/common_audio_c_gn/moz.build
--- a/media/webrtc/trunk/webrtc/common_audio/common_audio_c_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/common_audio/common_audio_c_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -140,12 +140,6 @@
     DEFINES["WEBRTC_POSIX"] = True
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
-    UNIFIED_SOURCES += [
-        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/complex_bit_reverse.c",
-        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/filter_ar_fast_q12.c",
-        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/spl_sqrt_floor.c"
-    ]
-
 if CONFIG["OS_TARGET"] == "NetBSD":
 
     DEFINES["USE_X11"] = "1"
@@ -218,6 +212,14 @@
         "/media/webrtc/trunk/webrtc/common_audio/signal_processing/spl_sqrt_floor_arm.S"
     ]
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    UNIFIED_SOURCES += [
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/complex_bit_reverse.c",
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/filter_ar_fast_q12.c",
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/spl_sqrt_floor.c"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -242,6 +244,14 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    UNIFIED_SOURCES += [
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/complex_bit_reverse.c",
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/filter_ar_fast_q12.c",
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/spl_sqrt_floor.c"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -278,6 +288,11 @@
         "/media/webrtc/trunk/webrtc/common_audio/signal_processing/spl_sqrt_floor.c"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CFLAGS += [
@@ -318,4 +333,20 @@
         "/media/webrtc/trunk/webrtc/common_audio/signal_processing/spl_sqrt_floor.c"
     ]
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    UNIFIED_SOURCES += [
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/complex_bit_reverse.c",
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/filter_ar_fast_q12.c",
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/spl_sqrt_floor.c"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    UNIFIED_SOURCES += [
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/complex_bit_reverse.c",
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/filter_ar_fast_q12.c",
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/spl_sqrt_floor.c"
+    ]
+
 Library("common_audio_c_gn")
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/common_audio/common_audio_gn/moz.build
--- a/media/webrtc/trunk/webrtc/common_audio/common_audio_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/common_audio/common_audio_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -212,6 +212,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/common_audio/common_audio_neon_c_gn/moz.build
--- a/media/webrtc/trunk/webrtc/common_audio/common_audio_neon_c_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/common_audio/common_audio_neon_c_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -39,7 +39,6 @@
     DEFINES["DYNAMIC_ANNOTATIONS_ENABLED"] = "0"
     DEFINES["NDEBUG"] = True
     DEFINES["NVALGRIND"] = True
-    DEFINES["_FORTIFY_SOURCE"] = "2"
 
 if CONFIG["MOZ_DEBUG"] == "1":
 
@@ -64,6 +63,12 @@
     DEFINES["USE_X11"] = "1"
     DEFINES["WEBRTC_BSD"] = True
 
+if CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["USE_NSS_CERTS"] = "1"
+    DEFINES["USE_X11"] = "1"
+    DEFINES["WEBRTC_LINUX"] = True
+
 if CONFIG["OS_TARGET"] == "NetBSD":
 
     DEFINES["USE_X11"] = "1"
@@ -87,4 +92,25 @@
     DEFINES["WEBRTC_ARCH_ARM"] = True
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
 
+if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
+
+    DEFINES["_FORTIFY_SOURCE"] = "2"
+
+if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "FreeBSD":
+
+    DEFINES["_FORTIFY_SOURCE"] = "2"
+
+if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "NetBSD":
+
+    DEFINES["_FORTIFY_SOURCE"] = "2"
+
+if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "OpenBSD":
+
+    DEFINES["_FORTIFY_SOURCE"] = "2"
+
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 Library("common_audio_neon_c_gn")
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/common_audio/common_audio_neon_gn/moz.build
--- a/media/webrtc/trunk/webrtc/common_audio/common_audio_neon_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/common_audio/common_audio_neon_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -38,7 +38,6 @@
     DEFINES["DYNAMIC_ANNOTATIONS_ENABLED"] = "0"
     DEFINES["NDEBUG"] = True
     DEFINES["NVALGRIND"] = True
-    DEFINES["_FORTIFY_SOURCE"] = "2"
 
 if CONFIG["MOZ_DEBUG"] == "1":
 
@@ -63,6 +62,12 @@
     DEFINES["USE_X11"] = "1"
     DEFINES["WEBRTC_BSD"] = True
 
+if CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["USE_NSS_CERTS"] = "1"
+    DEFINES["USE_X11"] = "1"
+    DEFINES["WEBRTC_LINUX"] = True
+
 if CONFIG["OS_TARGET"] == "NetBSD":
 
     DEFINES["USE_X11"] = "1"
@@ -86,4 +91,25 @@
     DEFINES["WEBRTC_ARCH_ARM"] = True
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
 
+if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
+
+    DEFINES["_FORTIFY_SOURCE"] = "2"
+
+if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "FreeBSD":
+
+    DEFINES["_FORTIFY_SOURCE"] = "2"
+
+if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "NetBSD":
+
+    DEFINES["_FORTIFY_SOURCE"] = "2"
+
+if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "OpenBSD":
+
+    DEFINES["_FORTIFY_SOURCE"] = "2"
+
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 Library("common_audio_neon_gn")
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/common_video/common_video_gn/moz.build
--- a/media/webrtc/trunk/webrtc/common_video/common_video_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/common_video/common_video_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -220,6 +220,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/logging/rtc_event_log_api_gn/moz.build
--- a/media/webrtc/trunk/webrtc/logging/rtc_event_log_api_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/logging/rtc_event_log_api_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -159,4 +159,9 @@
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 Library("rtc_event_log_api_gn")
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/logging/rtc_event_log_impl_gn/moz.build
--- a/media/webrtc/trunk/webrtc/logging/rtc_event_log_impl_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/logging/rtc_event_log_impl_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -202,6 +202,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/media/mozilla_rtc_media_gn/moz.build
--- a/media/webrtc/trunk/webrtc/media/mozilla_rtc_media_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/media/mozilla_rtc_media_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -181,6 +181,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_coding/audio_coding_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/audio_coding_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/audio_coding_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -116,12 +116,7 @@
     DEFINES["WEBRTC_POSIX"] = True
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
-    LOCAL_INCLUDES += [
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
-    ]
-
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -195,6 +190,16 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/include/"
     ]
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -219,6 +224,16 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -249,6 +264,15 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
@@ -281,4 +305,24 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
     ]
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("audio_coding_gn")
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_coding/audio_decoder_factory_interface_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/audio_decoder_factory_interface_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/audio_decoder_factory_interface_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -163,4 +163,9 @@
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 Library("audio_decoder_factory_interface_gn")
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_coding/audio_decoder_interface_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/audio_decoder_interface_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/audio_decoder_interface_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -184,6 +184,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_coding/audio_encoder_interface_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/audio_encoder_interface_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/audio_encoder_interface_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -183,6 +183,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_coding/audio_format_conversion_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/audio_format_conversion_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/audio_format_conversion_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -183,6 +183,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_coding/audio_format_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/audio_format_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/audio_format_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -179,6 +179,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_coding/audio_network_adaptor_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/audio_network_adaptor_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/audio_network_adaptor_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -203,6 +203,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_coding/builtin_audio_decoder_factory_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/builtin_audio_decoder_factory_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/builtin_audio_decoder_factory_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -104,12 +104,7 @@
     DEFINES["WEBRTC_POSIX"] = True
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
-    LOCAL_INCLUDES += [
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
-    ]
-
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -183,6 +178,16 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/include/"
     ]
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -207,6 +212,16 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -237,6 +252,15 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
@@ -269,4 +293,24 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
     ]
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("builtin_audio_decoder_factory_gn")
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_coding/cng_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/cng_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/cng_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -196,6 +196,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_coding/g711_c_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/g711_c_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/g711_c_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -180,6 +180,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_coding/g711_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/g711_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/g711_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -185,6 +185,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_coding/g722_c_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/g722_c_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/g722_c_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -184,6 +184,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_coding/g722_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/g722_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/g722_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -185,6 +185,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_coding/isac_c_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/isac_c_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/isac_c_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -224,6 +224,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_coding/isac_common_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/isac_common_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/isac_common_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -183,6 +183,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_coding/isac_fix_c_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/isac_fix_c_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/isac_fix_c_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -142,14 +142,6 @@
         "rt"
     ]
 
-    SOURCES += [
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/pitch_filter_c.c"
-    ]
-
-    UNIFIED_SOURCES += [
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/lattice_c.c"
-    ]
-
 if CONFIG["OS_TARGET"] == "NetBSD":
 
     DEFINES["USE_X11"] = "1"
@@ -229,6 +221,16 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/pitch_filter_armv6.S"
     ]
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    SOURCES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/pitch_filter_c.c"
+    ]
+
+    UNIFIED_SOURCES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/lattice_c.c"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -253,6 +255,16 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    SOURCES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/pitch_filter_c.c"
+    ]
+
+    UNIFIED_SOURCES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/lattice_c.c"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -295,6 +307,11 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/lattice_c.c"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CFLAGS += [
@@ -343,4 +360,24 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/lattice_c.c"
     ]
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    SOURCES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/pitch_filter_c.c"
+    ]
+
+    UNIFIED_SOURCES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/lattice_c.c"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    SOURCES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/pitch_filter_c.c"
+    ]
+
+    UNIFIED_SOURCES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/lattice_c.c"
+    ]
+
 Library("isac_fix_c_gn")
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_coding/isac_fix_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/isac_fix_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/isac_fix_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -197,6 +197,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_coding/isac_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/isac_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/isac_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -195,6 +195,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_coding/isac_neon_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/isac_neon_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/isac_neon_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -46,7 +46,6 @@
     DEFINES["DYNAMIC_ANNOTATIONS_ENABLED"] = "0"
     DEFINES["NDEBUG"] = True
     DEFINES["NVALGRIND"] = True
-    DEFINES["_FORTIFY_SOURCE"] = "2"
 
 if CONFIG["MOZ_DEBUG"] == "1":
 
@@ -75,6 +74,16 @@
     DEFINES["USE_X11"] = "1"
     DEFINES["WEBRTC_BSD"] = True
 
+if CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["USE_NSS_CERTS"] = "1"
+    DEFINES["USE_X11"] = "1"
+    DEFINES["WEBRTC_LINUX"] = True
+
+    OS_LIBS += [
+        "rt"
+    ]
+
 if CONFIG["OS_TARGET"] == "NetBSD":
 
     DEFINES["USE_X11"] = "1"
@@ -98,4 +107,25 @@
     DEFINES["WEBRTC_ARCH_ARM"] = True
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
 
+if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
+
+    DEFINES["_FORTIFY_SOURCE"] = "2"
+
+if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "FreeBSD":
+
+    DEFINES["_FORTIFY_SOURCE"] = "2"
+
+if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "NetBSD":
+
+    DEFINES["_FORTIFY_SOURCE"] = "2"
+
+if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "OpenBSD":
+
+    DEFINES["_FORTIFY_SOURCE"] = "2"
+
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 Library("isac_neon_gn")
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_coding/neteq_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/neteq_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/neteq_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -151,24 +151,11 @@
 
     DEFINES["USE_NSS_CERTS"] = "1"
     DEFINES["USE_X11"] = "1"
-    DEFINES["WEBRTC_CODEC_ISAC"] = True
     DEFINES["WEBRTC_LINUX"] = True
     DEFINES["WEBRTC_POSIX"] = True
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
-    LOCAL_INCLUDES += [
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
-    ]
-
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -264,6 +251,25 @@
         "/media/webrtc/trunk/"
     ]
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    DEFINES["WEBRTC_CODEC_ISAC"] = True
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -288,6 +294,25 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["WEBRTC_CODEC_ISAC"] = True
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -345,6 +370,15 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
@@ -413,4 +447,42 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
     ]
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["WEBRTC_CODEC_ISAC"] = True
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["WEBRTC_CODEC_ISAC"] = True
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("neteq_gn")
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_coding/pcm16b_c_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/pcm16b_c_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/pcm16b_c_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -180,6 +180,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_coding/pcm16b_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/pcm16b_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/pcm16b_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -186,6 +186,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_coding/rent_a_codec_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/rent_a_codec_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/rent_a_codec_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -105,12 +105,7 @@
     DEFINES["WEBRTC_POSIX"] = True
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
-    LOCAL_INCLUDES += [
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
-    ]
-
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -184,6 +179,16 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/include/"
     ]
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -208,6 +213,16 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -238,6 +253,15 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
@@ -270,4 +294,24 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
     ]
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("rent_a_codec_gn")
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_coding/webrtc_opus_c_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/webrtc_opus_c_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/webrtc_opus_c_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -184,6 +184,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_coding/webrtc_opus_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/webrtc_opus_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/webrtc_opus_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -196,6 +196,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_conference_mixer/audio_conference_mixer_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_conference_mixer/audio_conference_mixer_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_conference_mixer/audio_conference_mixer_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -196,6 +196,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_device/audio_device_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_device/audio_device_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_device/audio_device_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -119,7 +119,6 @@
     ]
 
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -187,6 +186,12 @@
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -211,6 +216,12 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -227,6 +238,15 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
@@ -239,4 +259,16 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("audio_device_gn")
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_mixer/audio_frame_manipulator_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_mixer/audio_frame_manipulator_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_mixer/audio_frame_manipulator_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -183,6 +183,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_mixer/audio_mixer_impl_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_mixer/audio_mixer_impl_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_mixer/audio_mixer_impl_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -193,6 +193,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_c_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_c_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_c_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -110,11 +110,6 @@
         "rt"
     ]
 
-    UNIFIED_SOURCES += [
-        "/media/webrtc/trunk/webrtc/modules/audio_processing/ns/noise_suppression.c",
-        "/media/webrtc/trunk/webrtc/modules/audio_processing/ns/ns_core.c"
-    ]
-
 if CONFIG["OS_TARGET"] == "NetBSD":
 
     DEFINES["USE_X11"] = "1"
@@ -190,6 +185,13 @@
         "/media/webrtc/trunk/webrtc/modules/audio_processing/ns/nsx_core_c.c"
     ]
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    UNIFIED_SOURCES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_processing/ns/noise_suppression.c",
+        "/media/webrtc/trunk/webrtc/modules/audio_processing/ns/ns_core.c"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -214,6 +216,13 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    UNIFIED_SOURCES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_processing/ns/noise_suppression.c",
+        "/media/webrtc/trunk/webrtc/modules/audio_processing/ns/ns_core.c"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -247,6 +256,11 @@
         "/media/webrtc/trunk/webrtc/modules/audio_processing/ns/ns_core.c"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CFLAGS += [
@@ -283,4 +297,18 @@
         "/media/webrtc/trunk/webrtc/modules/audio_processing/ns/ns_core.c"
     ]
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    UNIFIED_SOURCES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_processing/ns/noise_suppression.c",
+        "/media/webrtc/trunk/webrtc/modules/audio_processing/ns/ns_core.c"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    UNIFIED_SOURCES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_processing/ns/noise_suppression.c",
+        "/media/webrtc/trunk/webrtc/modules/audio_processing/ns/ns_core.c"
+    ]
+
 Library("audio_processing_c_gn")
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -164,7 +164,6 @@
     DEFINES["USE_NSS_CERTS"] = "1"
     DEFINES["USE_X11"] = "1"
     DEFINES["WEBRTC_LINUX"] = True
-    DEFINES["WEBRTC_NS_FLOAT"] = True
     DEFINES["WEBRTC_POSIX"] = True
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
@@ -234,6 +233,10 @@
     DEFINES["WEBRTC_HAS_NEON"] = True
     DEFINES["WEBRTC_NS_FIXED"] = True
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    DEFINES["WEBRTC_NS_FLOAT"] = True
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -258,6 +261,10 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["WEBRTC_NS_FLOAT"] = True
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -282,6 +289,11 @@
 
     DEFINES["WEBRTC_NS_FLOAT"] = True
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
@@ -306,4 +318,12 @@
 
     DEFINES["WEBRTC_NS_FLOAT"] = True
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["WEBRTC_NS_FLOAT"] = True
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["WEBRTC_NS_FLOAT"] = True
+
 Library("audio_processing_gn")
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_neon_c_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_neon_c_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_neon_c_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -37,7 +37,6 @@
     DEFINES["DYNAMIC_ANNOTATIONS_ENABLED"] = "0"
     DEFINES["NDEBUG"] = True
     DEFINES["NVALGRIND"] = True
-    DEFINES["_FORTIFY_SOURCE"] = "2"
 
 if CONFIG["MOZ_DEBUG"] == "1":
 
@@ -66,6 +65,16 @@
     DEFINES["USE_X11"] = "1"
     DEFINES["WEBRTC_BSD"] = True
 
+if CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["USE_NSS_CERTS"] = "1"
+    DEFINES["USE_X11"] = "1"
+    DEFINES["WEBRTC_LINUX"] = True
+
+    OS_LIBS += [
+        "rt"
+    ]
+
 if CONFIG["OS_TARGET"] == "NetBSD":
 
     DEFINES["USE_X11"] = "1"
@@ -89,4 +98,25 @@
     DEFINES["WEBRTC_ARCH_ARM"] = True
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
 
+if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
+
+    DEFINES["_FORTIFY_SOURCE"] = "2"
+
+if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "FreeBSD":
+
+    DEFINES["_FORTIFY_SOURCE"] = "2"
+
+if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "NetBSD":
+
+    DEFINES["_FORTIFY_SOURCE"] = "2"
+
+if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "OpenBSD":
+
+    DEFINES["_FORTIFY_SOURCE"] = "2"
+
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 Library("audio_processing_neon_c_gn")
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_neon_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_neon_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_neon_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -43,7 +43,6 @@
     DEFINES["DYNAMIC_ANNOTATIONS_ENABLED"] = "0"
     DEFINES["NDEBUG"] = True
     DEFINES["NVALGRIND"] = True
-    DEFINES["_FORTIFY_SOURCE"] = "2"
 
 if CONFIG["MOZ_DEBUG"] == "1":
 
@@ -72,6 +71,16 @@
     DEFINES["USE_X11"] = "1"
     DEFINES["WEBRTC_BSD"] = True
 
+if CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["USE_NSS_CERTS"] = "1"
+    DEFINES["USE_X11"] = "1"
+    DEFINES["WEBRTC_LINUX"] = True
+
+    OS_LIBS += [
+        "rt"
+    ]
+
 if CONFIG["OS_TARGET"] == "NetBSD":
 
     DEFINES["USE_X11"] = "1"
@@ -95,4 +104,25 @@
     DEFINES["WEBRTC_ARCH_ARM"] = True
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
 
+if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
+
+    DEFINES["_FORTIFY_SOURCE"] = "2"
+
+if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "FreeBSD":
+
+    DEFINES["_FORTIFY_SOURCE"] = "2"
+
+if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "NetBSD":
+
+    DEFINES["_FORTIFY_SOURCE"] = "2"
+
+if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "OpenBSD":
+
+    DEFINES["_FORTIFY_SOURCE"] = "2"
+
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 Library("audio_processing_neon_gn")
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/bitrate_controller/bitrate_controller_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/bitrate_controller/bitrate_controller_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/bitrate_controller/bitrate_controller_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -203,6 +203,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/congestion_controller/congestion_controller_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/congestion_controller/congestion_controller_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/congestion_controller/congestion_controller_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -109,7 +109,6 @@
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -173,6 +172,12 @@
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -197,6 +202,12 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -213,6 +224,15 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
@@ -225,4 +245,16 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("congestion_controller_gn")
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/desktop_capture/desktop_capture_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/desktop_capture/desktop_capture_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/desktop_capture/desktop_capture_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -346,6 +346,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/desktop_capture/primitives_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/desktop_capture/primitives_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/desktop_capture/primitives_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -158,6 +158,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/media_file/media_file_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/media_file/media_file_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/media_file/media_file_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -195,6 +195,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/pacing/pacing_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/pacing/pacing_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/pacing/pacing_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -204,6 +204,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/remote_bitrate_estimator/remote_bitrate_estimator_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/remote_bitrate_estimator/remote_bitrate_estimator_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/remote_bitrate_estimator/remote_bitrate_estimator_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -203,6 +203,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/rtp_rtcp/rtp_rtcp_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/rtp_rtcp/rtp_rtcp_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/rtp_rtcp/rtp_rtcp_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -274,6 +274,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/utility/utility_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/utility/utility_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/utility/utility_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -104,7 +104,6 @@
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -168,6 +167,12 @@
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -192,6 +197,12 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -208,6 +219,15 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
@@ -220,4 +240,16 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("utility_gn")
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/video_capture/video_capture_internal_impl_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/video_capture/video_capture_internal_impl_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/video_capture/video_capture_internal_impl_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -125,7 +125,6 @@
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -222,6 +221,12 @@
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -246,6 +251,12 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -262,6 +273,15 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
@@ -274,4 +294,16 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("video_capture_internal_impl_gn")
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/video_capture/video_capture_module_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/video_capture/video_capture_module_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/video_capture/video_capture_module_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -102,7 +102,6 @@
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -166,6 +165,12 @@
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -190,6 +195,12 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -206,6 +217,15 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
@@ -218,4 +238,16 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("video_capture_module_gn")
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/video_coding/video_coding_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/video_coding/video_coding_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/video_coding/video_coding_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -130,7 +130,6 @@
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -194,6 +193,12 @@
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -218,6 +223,12 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -234,6 +245,15 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
@@ -246,4 +266,16 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("video_coding_gn")
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/video_coding/video_coding_utility_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/video_coding/video_coding_utility_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/video_coding/video_coding_utility_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -210,6 +210,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/video_coding/webrtc_h264_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/video_coding/webrtc_h264_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/video_coding/webrtc_h264_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -201,6 +201,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/video_coding/webrtc_i420_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/video_coding/webrtc_i420_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/video_coding/webrtc_i420_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -203,6 +203,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/video_coding/webrtc_vp8_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/video_coding/webrtc_vp8_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/video_coding/webrtc_vp8_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -209,6 +209,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/video_coding/webrtc_vp9_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/video_coding/webrtc_vp9_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/video_coding/webrtc_vp9_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -205,6 +205,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/video_processing/video_processing_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/video_processing/video_processing_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/video_processing/video_processing_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -108,7 +108,6 @@
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -172,6 +171,12 @@
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -196,6 +201,12 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -212,6 +223,15 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
@@ -224,4 +244,16 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("video_processing_gn")
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/modules/video_processing/video_processing_neon_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/video_processing/video_processing_neon_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/video_processing/video_processing_neon_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -37,7 +37,6 @@
     DEFINES["DYNAMIC_ANNOTATIONS_ENABLED"] = "0"
     DEFINES["NDEBUG"] = True
     DEFINES["NVALGRIND"] = True
-    DEFINES["_FORTIFY_SOURCE"] = "2"
 
 if CONFIG["MOZ_DEBUG"] == "1":
 
@@ -62,6 +61,12 @@
     DEFINES["USE_X11"] = "1"
     DEFINES["WEBRTC_BSD"] = True
 
+if CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["USE_NSS_CERTS"] = "1"
+    DEFINES["USE_X11"] = "1"
+    DEFINES["WEBRTC_LINUX"] = True
+
 if CONFIG["OS_TARGET"] == "NetBSD":
 
     DEFINES["USE_X11"] = "1"
@@ -85,4 +90,25 @@
     DEFINES["WEBRTC_ARCH_ARM"] = True
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
 
+if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
+
+    DEFINES["_FORTIFY_SOURCE"] = "2"
+
+if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "FreeBSD":
+
+    DEFINES["_FORTIFY_SOURCE"] = "2"
+
+if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "NetBSD":
+
+    DEFINES["_FORTIFY_SOURCE"] = "2"
+
+if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "OpenBSD":
+
+    DEFINES["_FORTIFY_SOURCE"] = "2"
+
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 Library("video_processing_neon_gn")
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/system_wrappers/field_trial_default_gn/moz.build
--- a/media/webrtc/trunk/webrtc/system_wrappers/field_trial_default_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/system_wrappers/field_trial_default_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -179,6 +179,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/system_wrappers/metrics_default_gn/moz.build
--- a/media/webrtc/trunk/webrtc/system_wrappers/metrics_default_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/system_wrappers/metrics_default_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -179,6 +179,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/system_wrappers/system_wrappers_gn/moz.build
--- a/media/webrtc/trunk/webrtc/system_wrappers/system_wrappers_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/system_wrappers/system_wrappers_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -274,6 +274,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/video/video_gn/moz.build
--- a/media/webrtc/trunk/webrtc/video/video_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/video/video_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -227,6 +227,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/video_engine/video_engine_gn/moz.build
--- a/media/webrtc/trunk/webrtc/video_engine/video_engine_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/video_engine/video_engine_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -154,6 +154,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/voice_engine/audio_coder_gn/moz.build
--- a/media/webrtc/trunk/webrtc/voice_engine/audio_coder_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/voice_engine/audio_coder_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -92,7 +92,6 @@
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -154,6 +153,12 @@
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -178,6 +183,12 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -194,6 +205,15 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
@@ -206,4 +226,16 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("audio_coder_gn")
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/voice_engine/file_player_gn/moz.build
--- a/media/webrtc/trunk/webrtc/voice_engine/file_player_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/voice_engine/file_player_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -93,7 +93,6 @@
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -155,6 +154,12 @@
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -179,6 +184,12 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -195,6 +206,15 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
@@ -207,4 +227,16 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("file_player_gn")
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/voice_engine/file_recorder_gn/moz.build
--- a/media/webrtc/trunk/webrtc/voice_engine/file_recorder_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/voice_engine/file_recorder_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -93,7 +93,6 @@
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -155,6 +154,12 @@
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -179,6 +184,12 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
@@ -195,6 +206,15 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
@@ -207,4 +227,16 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("file_recorder_gn")
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/voice_engine/level_indicator_gn/moz.build
--- a/media/webrtc/trunk/webrtc/voice_engine/level_indicator_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/voice_engine/level_indicator_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -194,6 +194,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/voice_engine/voice_engine_gn/moz.build
--- a/media/webrtc/trunk/webrtc/voice_engine/voice_engine_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/voice_engine/voice_engine_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -233,6 +233,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/webrtc_common_gn/moz.build
--- a/media/webrtc/trunk/webrtc/webrtc_common_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/webrtc_common_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -180,6 +180,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
diff -r 681eb7dfa324 media/webrtc/trunk/webrtc/webrtc_gn/moz.build
--- a/media/webrtc/trunk/webrtc/webrtc_gn/moz.build	Wed Jun 20 14:28:15 2018 +0300
+++ b/media/webrtc/trunk/webrtc/webrtc_gn/moz.build	Wed Jun 20 19:57:47 2018 +0200
@@ -284,6 +284,11 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "NetBSD":
 
     CXXFLAGS += [
