diff -r d49587f5ccd3 media/webrtc/trunk/moz.build
--- a/media/webrtc/trunk/moz.build	Sun Oct 14 12:25:38 2018 +0300
+++ b/media/webrtc/trunk/moz.build	Sun Oct 14 13:30:57 2018 +0200
@@ -121,7 +121,6 @@
     DIRS += [
         "/media/webrtc/trunk/webrtc/modules/desktop_capture/desktop_capture_gn",
         "/media/webrtc/trunk/webrtc/modules/desktop_capture/primitives_gn",
-        "/media/webrtc/trunk/webrtc/system_wrappers/cpu_features_linux_gn",
         "/media/webrtc/trunk/webrtc/video_engine/video_engine_gn"
     ]
 
@@ -239,9 +238,27 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/isac_neon_gn",
         "/media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_neon_c_gn",
         "/media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_neon_gn",
+        "/media/webrtc/trunk/webrtc/modules/video_processing/video_processing_neon_gn",
+        "/media/webrtc/trunk/webrtc/system_wrappers/cpu_features_linux_gn"
+    ]
+
+if CONFIG["CPU_ARCH"] == "arm" and CONFIG["OS_TARGET"] == "Linux":
+
+    DIRS += [
+        "/media/webrtc/trunk/webrtc/common_audio/common_audio_neon_c_gn",
+        "/media/webrtc/trunk/webrtc/common_audio/common_audio_neon_gn",
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/isac_neon_gn",
+        "/media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_neon_c_gn",
+        "/media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_neon_gn",
         "/media/webrtc/trunk/webrtc/modules/video_processing/video_processing_neon_gn"
     ]
 
+if CONFIG["CPU_ARCH"] == "ppc64" and CONFIG["OS_TARGET"] == "Linux":
+
+    DIRS += [
+        "/media/webrtc/trunk/webrtc/system_wrappers/cpu_features_linux_gn"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "Linux":
 
     DIRS += [
diff -r d49587f5ccd3 media/webrtc/trunk/webrtc/common_audio/common_audio_c_gn/moz.build
--- a/media/webrtc/trunk/webrtc/common_audio/common_audio_c_gn/moz.build	Sun Oct 14 12:25:38 2018 +0300
+++ b/media/webrtc/trunk/webrtc/common_audio/common_audio_c_gn/moz.build	Sun Oct 14 13:30:57 2018 +0200
@@ -140,12 +140,6 @@
     DEFINES["WEBRTC_POSIX"] = True
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
-    UNIFIED_SOURCES += [
-        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/complex_bit_reverse.c",
-        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/filter_ar_fast_q12.c",
-        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/spl_sqrt_floor.c"
-    ]
-
 if CONFIG["OS_TARGET"] == "NetBSD":
 
     DEFINES["USE_X11"] = "1"
@@ -196,6 +190,12 @@
     DEFINES["WEBRTC_ARCH_ARM64"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+    UNIFIED_SOURCES += [
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/complex_bit_reverse.c",
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/filter_ar_fast_q12.c",
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/spl_sqrt_floor.c"
+    ]
+
 if CONFIG["CPU_ARCH"] == "arm":
 
     CFLAGS += [
@@ -212,6 +212,14 @@
         "/media/webrtc/trunk/webrtc/common_audio/signal_processing/spl_sqrt_floor_arm.S"
     ]
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    UNIFIED_SOURCES += [
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/complex_bit_reverse.c",
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/filter_ar_fast_q12.c",
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/spl_sqrt_floor.c"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -236,14 +244,6 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Android":
-
-    UNIFIED_SOURCES += [
-        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/complex_bit_reverse.c",
-        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/filter_ar_fast_q12.c",
-        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/spl_sqrt_floor.c"
-    ]
-
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "Android":
 
     CFLAGS += [
@@ -264,14 +264,6 @@
         "/media/webrtc/trunk/webrtc/common_audio/signal_processing/spl_sqrt_floor.c"
     ]
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "FreeBSD":
-
-    UNIFIED_SOURCES += [
-        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/complex_bit_reverse.c",
-        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/filter_ar_fast_q12.c",
-        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/spl_sqrt_floor.c"
-    ]
-
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "FreeBSD":
 
     CFLAGS += [
@@ -297,7 +289,15 @@
     DEFINES["DISABLE_NACL"] = True
     DEFINES["NO_TCMALLOC"] = True
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "NetBSD":
+if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "Linux":
+
+    UNIFIED_SOURCES += [
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/complex_bit_reverse.c",
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/filter_ar_fast_q12.c",
+        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/spl_sqrt_floor.c"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and CONFIG["OS_TARGET"] == "Linux":
 
     UNIFIED_SOURCES += [
         "/media/webrtc/trunk/webrtc/common_audio/signal_processing/complex_bit_reverse.c",
@@ -325,14 +325,6 @@
         "/media/webrtc/trunk/webrtc/common_audio/signal_processing/spl_sqrt_floor.c"
     ]
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "OpenBSD":
-
-    UNIFIED_SOURCES += [
-        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/complex_bit_reverse.c",
-        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/filter_ar_fast_q12.c",
-        "/media/webrtc/trunk/webrtc/common_audio/signal_processing/spl_sqrt_floor.c"
-    ]
-
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "OpenBSD":
 
     CFLAGS += [
diff -r d49587f5ccd3 media/webrtc/trunk/webrtc/common_audio/common_audio_neon_c_gn/moz.build
--- a/media/webrtc/trunk/webrtc/common_audio/common_audio_neon_c_gn/moz.build	Sun Oct 14 12:25:38 2018 +0300
+++ b/media/webrtc/trunk/webrtc/common_audio/common_audio_neon_c_gn/moz.build	Sun Oct 14 13:30:57 2018 +0200
@@ -65,11 +65,8 @@
 
 if CONFIG["OS_TARGET"] == "Linux":
 
-    DEFINES["DISABLE_NACL"] = True
-    DEFINES["NO_TCMALLOC"] = True
     DEFINES["USE_NSS_CERTS"] = "1"
     DEFINES["USE_X11"] = "1"
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
     DEFINES["WEBRTC_LINUX"] = True
 
 if CONFIG["OS_TARGET"] == "NetBSD":
@@ -82,6 +79,10 @@
     DEFINES["USE_X11"] = "1"
     DEFINES["WEBRTC_BSD"] = True
 
+if CONFIG["CPU_ARCH"] == "aarch64":
+
+    DEFINES["WEBRTC_ARCH_ARM64"] = True
+
 if CONFIG["CPU_ARCH"] == "arm":
 
     CFLAGS += [
@@ -107,20 +108,14 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Android":
-
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "FreeBSD":
-
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "NetBSD":
-
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
+if CONFIG["CPU_ARCH"] == "aarch64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "OpenBSD":
-
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
 
 Library("common_audio_neon_c_gn")
diff -r d49587f5ccd3 media/webrtc/trunk/webrtc/common_audio/common_audio_neon_gn/moz.build
--- a/media/webrtc/trunk/webrtc/common_audio/common_audio_neon_gn/moz.build	Sun Oct 14 12:25:38 2018 +0300
+++ b/media/webrtc/trunk/webrtc/common_audio/common_audio_neon_gn/moz.build	Sun Oct 14 13:30:57 2018 +0200
@@ -64,11 +64,8 @@
 
 if CONFIG["OS_TARGET"] == "Linux":
 
-    DEFINES["DISABLE_NACL"] = True
-    DEFINES["NO_TCMALLOC"] = True
     DEFINES["USE_NSS_CERTS"] = "1"
     DEFINES["USE_X11"] = "1"
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
     DEFINES["WEBRTC_LINUX"] = True
 
 if CONFIG["OS_TARGET"] == "NetBSD":
@@ -81,6 +78,10 @@
     DEFINES["USE_X11"] = "1"
     DEFINES["WEBRTC_BSD"] = True
 
+if CONFIG["CPU_ARCH"] == "aarch64":
+
+    DEFINES["WEBRTC_ARCH_ARM64"] = True
+
 if CONFIG["CPU_ARCH"] == "arm":
 
     CXXFLAGS += [
@@ -106,20 +107,14 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Android":
-
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "FreeBSD":
-
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "NetBSD":
-
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
+if CONFIG["CPU_ARCH"] == "aarch64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "OpenBSD":
-
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
 
 Library("common_audio_neon_gn")
diff -r d49587f5ccd3 media/webrtc/trunk/webrtc/modules/audio_coding/audio_coding_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/audio_coding_gn/moz.build	Sun Oct 14 12:25:38 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/audio_coding_gn/moz.build	Sun Oct 14 13:30:57 2018 +0200
@@ -116,12 +116,7 @@
     DEFINES["WEBRTC_POSIX"] = True
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
-    LOCAL_INCLUDES += [
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
-    ]
-
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -177,6 +172,10 @@
     DEFINES["WEBRTC_ARCH_ARM64"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
 if CONFIG["CPU_ARCH"] == "arm":
 
     CXXFLAGS += [
@@ -191,6 +190,16 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/include/"
     ]
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -215,10 +224,10 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Android":
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
 
-    LOCAL_INCLUDES += [
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    OS_LIBS += [
+        "m"
     ]
 
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "Android":
@@ -237,12 +246,6 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
     ]
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "FreeBSD":
-
-    LOCAL_INCLUDES += [
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
-    ]
-
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "FreeBSD":
 
     CXXFLAGS += [
@@ -264,7 +267,13 @@
     DEFINES["DISABLE_NACL"] = True
     DEFINES["NO_TCMALLOC"] = True
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "NetBSD":
+if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "Linux":
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and CONFIG["OS_TARGET"] == "Linux":
 
     LOCAL_INCLUDES += [
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
@@ -286,12 +295,6 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
     ]
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "OpenBSD":
-
-    LOCAL_INCLUDES += [
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
-    ]
-
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "OpenBSD":
 
     CXXFLAGS += [
@@ -308,4 +311,22 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("audio_coding_gn")
diff -r d49587f5ccd3 media/webrtc/trunk/webrtc/modules/audio_coding/builtin_audio_decoder_factory_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/builtin_audio_decoder_factory_gn/moz.build	Sun Oct 14 12:25:38 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/builtin_audio_decoder_factory_gn/moz.build	Sun Oct 14 13:30:57 2018 +0200
@@ -104,12 +104,7 @@
     DEFINES["WEBRTC_POSIX"] = True
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
-    LOCAL_INCLUDES += [
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
-    ]
-
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -165,6 +160,10 @@
     DEFINES["WEBRTC_ARCH_ARM64"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
 if CONFIG["CPU_ARCH"] == "arm":
 
     CXXFLAGS += [
@@ -179,6 +178,16 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/include/"
     ]
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -203,10 +212,10 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Android":
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
 
-    LOCAL_INCLUDES += [
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    OS_LIBS += [
+        "m"
     ]
 
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "Android":
@@ -225,12 +234,6 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
     ]
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "FreeBSD":
-
-    LOCAL_INCLUDES += [
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
-    ]
-
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "FreeBSD":
 
     CXXFLAGS += [
@@ -252,7 +255,13 @@
     DEFINES["DISABLE_NACL"] = True
     DEFINES["NO_TCMALLOC"] = True
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "NetBSD":
+if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "Linux":
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and CONFIG["OS_TARGET"] == "Linux":
 
     LOCAL_INCLUDES += [
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
@@ -274,12 +283,6 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
     ]
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "OpenBSD":
-
-    LOCAL_INCLUDES += [
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
-    ]
-
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "OpenBSD":
 
     CXXFLAGS += [
@@ -296,4 +299,22 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("builtin_audio_decoder_factory_gn")
diff -r d49587f5ccd3 media/webrtc/trunk/webrtc/modules/audio_coding/isac_fix_c_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/isac_fix_c_gn/moz.build	Sun Oct 14 12:25:38 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/isac_fix_c_gn/moz.build	Sun Oct 14 13:30:57 2018 +0200
@@ -142,14 +142,6 @@
         "rt"
     ]
 
-    SOURCES += [
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/pitch_filter_c.c"
-    ]
-
-    UNIFIED_SOURCES += [
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/lattice_c.c"
-    ]
-
 if CONFIG["OS_TARGET"] == "NetBSD":
 
     DEFINES["USE_X11"] = "1"
@@ -206,6 +198,14 @@
     DEFINES["WEBRTC_ARCH_ARM64"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+    SOURCES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/pitch_filter_c.c"
+    ]
+
+    UNIFIED_SOURCES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/lattice_c.c"
+    ]
+
 if CONFIG["CPU_ARCH"] == "arm":
 
     CFLAGS += [
@@ -221,6 +221,16 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/pitch_filter_armv6.S"
     ]
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    SOURCES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/pitch_filter_c.c"
+    ]
+
+    UNIFIED_SOURCES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/lattice_c.c"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -245,16 +255,6 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Android":
-
-    SOURCES += [
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/pitch_filter_c.c"
-    ]
-
-    UNIFIED_SOURCES += [
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/lattice_c.c"
-    ]
-
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "Android":
 
     CFLAGS += [
@@ -279,16 +279,6 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/lattice_c.c"
     ]
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "FreeBSD":
-
-    SOURCES += [
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/pitch_filter_c.c"
-    ]
-
-    UNIFIED_SOURCES += [
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/lattice_c.c"
-    ]
-
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "FreeBSD":
 
     CFLAGS += [
@@ -318,7 +308,17 @@
     DEFINES["DISABLE_NACL"] = True
     DEFINES["NO_TCMALLOC"] = True
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "NetBSD":
+if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "Linux":
+
+    SOURCES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/pitch_filter_c.c"
+    ]
+
+    UNIFIED_SOURCES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/lattice_c.c"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and CONFIG["OS_TARGET"] == "Linux":
 
     SOURCES += [
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/pitch_filter_c.c"
@@ -352,16 +352,6 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/lattice_c.c"
     ]
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "OpenBSD":
-
-    SOURCES += [
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/pitch_filter_c.c"
-    ]
-
-    UNIFIED_SOURCES += [
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/source/lattice_c.c"
-    ]
-
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "OpenBSD":
 
     CFLAGS += [
diff -r d49587f5ccd3 media/webrtc/trunk/webrtc/modules/audio_coding/isac_neon_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/isac_neon_gn/moz.build	Sun Oct 14 12:25:38 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/isac_neon_gn/moz.build	Sun Oct 14 13:30:57 2018 +0200
@@ -76,11 +76,8 @@
 
 if CONFIG["OS_TARGET"] == "Linux":
 
-    DEFINES["DISABLE_NACL"] = True
-    DEFINES["NO_TCMALLOC"] = True
     DEFINES["USE_NSS_CERTS"] = "1"
     DEFINES["USE_X11"] = "1"
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
     DEFINES["WEBRTC_LINUX"] = True
 
     OS_LIBS += [
@@ -97,6 +94,10 @@
     DEFINES["USE_X11"] = "1"
     DEFINES["WEBRTC_BSD"] = True
 
+if CONFIG["CPU_ARCH"] == "aarch64":
+
+    DEFINES["WEBRTC_ARCH_ARM64"] = True
+
 if CONFIG["CPU_ARCH"] == "arm":
 
     CFLAGS += [
@@ -122,20 +123,14 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Android":
-
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "FreeBSD":
-
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "NetBSD":
-
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
+if CONFIG["CPU_ARCH"] == "aarch64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "OpenBSD":
-
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
 
 Library("isac_neon_gn")
diff -r d49587f5ccd3 media/webrtc/trunk/webrtc/modules/audio_coding/neteq_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/neteq_gn/moz.build	Sun Oct 14 12:25:38 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/neteq_gn/moz.build	Sun Oct 14 13:30:57 2018 +0200
@@ -151,24 +151,11 @@
 
     DEFINES["USE_NSS_CERTS"] = "1"
     DEFINES["USE_X11"] = "1"
-    DEFINES["WEBRTC_CODEC_ISAC"] = True
     DEFINES["WEBRTC_LINUX"] = True
     DEFINES["WEBRTC_POSIX"] = True
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
-    LOCAL_INCLUDES += [
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
-    ]
-
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -230,8 +217,20 @@
 if CONFIG["CPU_ARCH"] == "aarch64":
 
     DEFINES["WEBRTC_ARCH_ARM64"] = True
+    DEFINES["WEBRTC_CODEC_ISAC"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
 if CONFIG["CPU_ARCH"] == "arm":
 
     CXXFLAGS += [
@@ -252,6 +251,25 @@
         "/media/webrtc/trunk/"
     ]
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    DEFINES["WEBRTC_CODEC_ISAC"] = True
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -276,19 +294,10 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Android":
-
-    DEFINES["WEBRTC_CODEC_ISAC"] = True
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
 
-    LOCAL_INCLUDES += [
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    OS_LIBS += [
+        "m"
     ]
 
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "Android":
@@ -325,21 +334,6 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
     ]
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "FreeBSD":
-
-    DEFINES["WEBRTC_CODEC_ISAC"] = True
-
-    LOCAL_INCLUDES += [
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
-    ]
-
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "FreeBSD":
 
     CXXFLAGS += [
@@ -379,7 +373,22 @@
     DEFINES["DISABLE_NACL"] = True
     DEFINES["NO_TCMALLOC"] = True
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "NetBSD":
+if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "Linux":
+
+    DEFINES["WEBRTC_CODEC_ISAC"] = True
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/",
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and CONFIG["OS_TARGET"] == "Linux":
 
     DEFINES["WEBRTC_CODEC_ISAC"] = True
 
@@ -428,21 +437,6 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
     ]
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "OpenBSD":
-
-    DEFINES["WEBRTC_CODEC_ISAC"] = True
-
-    LOCAL_INCLUDES += [
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/",
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
-    ]
-
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "OpenBSD":
 
     CXXFLAGS += [
@@ -477,4 +471,22 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("neteq_gn")
diff -r d49587f5ccd3 media/webrtc/trunk/webrtc/modules/audio_coding/rent_a_codec_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_coding/rent_a_codec_gn/moz.build	Sun Oct 14 12:25:38 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_coding/rent_a_codec_gn/moz.build	Sun Oct 14 13:30:57 2018 +0200
@@ -105,12 +105,7 @@
     DEFINES["WEBRTC_POSIX"] = True
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
-    LOCAL_INCLUDES += [
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
-    ]
-
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -166,6 +161,10 @@
     DEFINES["WEBRTC_ARCH_ARM64"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
 if CONFIG["CPU_ARCH"] == "arm":
 
     CXXFLAGS += [
@@ -180,6 +179,16 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/fix/include/"
     ]
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -204,10 +213,10 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Android":
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
 
-    LOCAL_INCLUDES += [
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    OS_LIBS += [
+        "m"
     ]
 
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "Android":
@@ -226,12 +235,6 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
     ]
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "FreeBSD":
-
-    LOCAL_INCLUDES += [
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
-    ]
-
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "FreeBSD":
 
     CXXFLAGS += [
@@ -253,7 +256,13 @@
     DEFINES["DISABLE_NACL"] = True
     DEFINES["NO_TCMALLOC"] = True
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "NetBSD":
+if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "Linux":
+
+    LOCAL_INCLUDES += [
+        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and CONFIG["OS_TARGET"] == "Linux":
 
     LOCAL_INCLUDES += [
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
@@ -275,12 +284,6 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
     ]
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "OpenBSD":
-
-    LOCAL_INCLUDES += [
-        "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
-    ]
-
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "OpenBSD":
 
     CXXFLAGS += [
@@ -297,4 +300,22 @@
         "/media/webrtc/trunk/webrtc/modules/audio_coding/codecs/isac/main/include/"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("rent_a_codec_gn")
diff -r d49587f5ccd3 media/webrtc/trunk/webrtc/modules/audio_device/audio_device_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_device/audio_device_gn/moz.build	Sun Oct 14 12:25:38 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_device/audio_device_gn/moz.build	Sun Oct 14 13:30:57 2018 +0200
@@ -119,7 +119,6 @@
     ]
 
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -187,6 +186,12 @@
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -211,6 +216,12 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "Android":
 
     CXXFLAGS += [
@@ -240,4 +251,22 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("audio_device_gn")
diff -r d49587f5ccd3 media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_neon_c_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_neon_c_gn/moz.build	Sun Oct 14 12:25:38 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_neon_c_gn/moz.build	Sun Oct 14 13:30:57 2018 +0200
@@ -67,11 +67,8 @@
 
 if CONFIG["OS_TARGET"] == "Linux":
 
-    DEFINES["DISABLE_NACL"] = True
-    DEFINES["NO_TCMALLOC"] = True
     DEFINES["USE_NSS_CERTS"] = "1"
     DEFINES["USE_X11"] = "1"
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
     DEFINES["WEBRTC_LINUX"] = True
 
     OS_LIBS += [
@@ -88,6 +85,10 @@
     DEFINES["USE_X11"] = "1"
     DEFINES["WEBRTC_BSD"] = True
 
+if CONFIG["CPU_ARCH"] == "aarch64":
+
+    DEFINES["WEBRTC_ARCH_ARM64"] = True
+
 if CONFIG["CPU_ARCH"] == "arm":
 
     CFLAGS += [
@@ -113,20 +114,14 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Android":
-
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "FreeBSD":
-
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "NetBSD":
-
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
+if CONFIG["CPU_ARCH"] == "aarch64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "OpenBSD":
-
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
 
 Library("audio_processing_neon_c_gn")
diff -r d49587f5ccd3 media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_neon_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_neon_gn/moz.build	Sun Oct 14 12:25:38 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/audio_processing/audio_processing_neon_gn/moz.build	Sun Oct 14 13:30:57 2018 +0200
@@ -73,11 +73,8 @@
 
 if CONFIG["OS_TARGET"] == "Linux":
 
-    DEFINES["DISABLE_NACL"] = True
-    DEFINES["NO_TCMALLOC"] = True
     DEFINES["USE_NSS_CERTS"] = "1"
     DEFINES["USE_X11"] = "1"
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
     DEFINES["WEBRTC_LINUX"] = True
 
     OS_LIBS += [
@@ -94,6 +91,10 @@
     DEFINES["USE_X11"] = "1"
     DEFINES["WEBRTC_BSD"] = True
 
+if CONFIG["CPU_ARCH"] == "aarch64":
+
+    DEFINES["WEBRTC_ARCH_ARM64"] = True
+
 if CONFIG["CPU_ARCH"] == "arm":
 
     CXXFLAGS += [
@@ -119,20 +120,14 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Android":
-
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "FreeBSD":
-
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "NetBSD":
-
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
+if CONFIG["CPU_ARCH"] == "aarch64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "OpenBSD":
-
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
 
 Library("audio_processing_neon_gn")
diff -r d49587f5ccd3 media/webrtc/trunk/webrtc/modules/congestion_controller/congestion_controller_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/congestion_controller/congestion_controller_gn/moz.build	Sun Oct 14 12:25:38 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/congestion_controller/congestion_controller_gn/moz.build	Sun Oct 14 13:30:57 2018 +0200
@@ -109,7 +109,6 @@
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -173,6 +172,12 @@
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -197,6 +202,12 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "Android":
 
     CXXFLAGS += [
@@ -226,4 +237,22 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("congestion_controller_gn")
diff -r d49587f5ccd3 media/webrtc/trunk/webrtc/modules/utility/utility_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/utility/utility_gn/moz.build	Sun Oct 14 12:25:38 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/utility/utility_gn/moz.build	Sun Oct 14 13:30:57 2018 +0200
@@ -104,7 +104,6 @@
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -168,6 +167,12 @@
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -192,6 +197,12 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "Android":
 
     CXXFLAGS += [
@@ -221,4 +232,22 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("utility_gn")
diff -r d49587f5ccd3 media/webrtc/trunk/webrtc/modules/video_capture/video_capture_internal_impl_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/video_capture/video_capture_internal_impl_gn/moz.build	Sun Oct 14 12:25:38 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/video_capture/video_capture_internal_impl_gn/moz.build	Sun Oct 14 13:30:57 2018 +0200
@@ -125,7 +125,6 @@
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -222,6 +221,12 @@
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -246,6 +251,12 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "Android":
 
     CXXFLAGS += [
@@ -275,4 +286,22 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("video_capture_internal_impl_gn")
diff -r d49587f5ccd3 media/webrtc/trunk/webrtc/modules/video_capture/video_capture_module_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/video_capture/video_capture_module_gn/moz.build	Sun Oct 14 12:25:38 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/video_capture/video_capture_module_gn/moz.build	Sun Oct 14 13:30:57 2018 +0200
@@ -102,7 +102,6 @@
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -166,6 +165,12 @@
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -190,6 +195,12 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "Android":
 
     CXXFLAGS += [
@@ -219,4 +230,22 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("video_capture_module_gn")
diff -r d49587f5ccd3 media/webrtc/trunk/webrtc/modules/video_coding/video_coding_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/video_coding/video_coding_gn/moz.build	Sun Oct 14 12:25:38 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/video_coding/video_coding_gn/moz.build	Sun Oct 14 13:30:57 2018 +0200
@@ -130,7 +130,6 @@
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -194,6 +193,12 @@
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -218,6 +223,12 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "Android":
 
     CXXFLAGS += [
@@ -247,4 +258,22 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("video_coding_gn")
diff -r d49587f5ccd3 media/webrtc/trunk/webrtc/modules/video_processing/video_processing_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/video_processing/video_processing_gn/moz.build	Sun Oct 14 12:25:38 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/video_processing/video_processing_gn/moz.build	Sun Oct 14 13:30:57 2018 +0200
@@ -108,7 +108,6 @@
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -172,6 +171,12 @@
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -196,6 +201,12 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "Android":
 
     CXXFLAGS += [
@@ -225,4 +236,22 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("video_processing_gn")
diff -r d49587f5ccd3 media/webrtc/trunk/webrtc/modules/video_processing/video_processing_neon_gn/moz.build
--- a/media/webrtc/trunk/webrtc/modules/video_processing/video_processing_neon_gn/moz.build	Sun Oct 14 12:25:38 2018 +0300
+++ b/media/webrtc/trunk/webrtc/modules/video_processing/video_processing_neon_gn/moz.build	Sun Oct 14 13:30:57 2018 +0200
@@ -63,11 +63,8 @@
 
 if CONFIG["OS_TARGET"] == "Linux":
 
-    DEFINES["DISABLE_NACL"] = True
-    DEFINES["NO_TCMALLOC"] = True
     DEFINES["USE_NSS_CERTS"] = "1"
     DEFINES["USE_X11"] = "1"
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
     DEFINES["WEBRTC_LINUX"] = True
 
 if CONFIG["OS_TARGET"] == "NetBSD":
@@ -80,6 +77,10 @@
     DEFINES["USE_X11"] = "1"
     DEFINES["WEBRTC_BSD"] = True
 
+if CONFIG["CPU_ARCH"] == "aarch64":
+
+    DEFINES["WEBRTC_ARCH_ARM64"] = True
+
 if CONFIG["CPU_ARCH"] == "arm":
 
     CXXFLAGS += [
@@ -105,20 +106,14 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "Android":
-
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "FreeBSD":
-
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "NetBSD":
-
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
+if CONFIG["CPU_ARCH"] == "aarch64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
 
-if CONFIG["CPU_ARCH"] == "aarch64" and CONFIG["OS_TARGET"] == "OpenBSD":
-
-    DEFINES["WEBRTC_ARCH_ARM64"] = True
+    DEFINES["DISABLE_NACL"] = True
+    DEFINES["NO_TCMALLOC"] = True
 
 Library("video_processing_neon_gn")
diff -r d49587f5ccd3 media/webrtc/trunk/webrtc/voice_engine/audio_coder_gn/moz.build
--- a/media/webrtc/trunk/webrtc/voice_engine/audio_coder_gn/moz.build	Sun Oct 14 12:25:38 2018 +0300
+++ b/media/webrtc/trunk/webrtc/voice_engine/audio_coder_gn/moz.build	Sun Oct 14 13:30:57 2018 +0200
@@ -92,7 +92,6 @@
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -154,6 +153,12 @@
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -178,6 +183,12 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "Android":
 
     CXXFLAGS += [
@@ -207,4 +218,22 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("audio_coder_gn")
diff -r d49587f5ccd3 media/webrtc/trunk/webrtc/voice_engine/file_player_gn/moz.build
--- a/media/webrtc/trunk/webrtc/voice_engine/file_player_gn/moz.build	Sun Oct 14 12:25:38 2018 +0300
+++ b/media/webrtc/trunk/webrtc/voice_engine/file_player_gn/moz.build	Sun Oct 14 13:30:57 2018 +0200
@@ -93,7 +93,6 @@
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -155,6 +154,12 @@
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -179,6 +184,12 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "Android":
 
     CXXFLAGS += [
@@ -208,4 +219,22 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("file_player_gn")
diff -r d49587f5ccd3 media/webrtc/trunk/webrtc/voice_engine/file_recorder_gn/moz.build
--- a/media/webrtc/trunk/webrtc/voice_engine/file_recorder_gn/moz.build	Sun Oct 14 12:25:38 2018 +0300
+++ b/media/webrtc/trunk/webrtc/voice_engine/file_recorder_gn/moz.build	Sun Oct 14 13:30:57 2018 +0200
@@ -93,7 +93,6 @@
     DEFINES["_FILE_OFFSET_BITS"] = "64"
 
     OS_LIBS += [
-        "m",
         "rt"
     ]
 
@@ -155,6 +154,12 @@
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+if CONFIG["CPU_ARCH"] == "ppc64":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Android":
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
@@ -179,6 +184,12 @@
 
     DEFINES["_FORTIFY_SOURCE"] = "2"
 
+if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 if CONFIG["CPU_ARCH"] == "x86" and CONFIG["OS_TARGET"] == "Android":
 
     CXXFLAGS += [
@@ -208,4 +219,22 @@
         "-msse2"
     ]
 
+if CONFIG["CPU_ARCH"] == "aarch64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
+if CONFIG["CPU_ARCH"] == "x86_64" and not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "Linux":
+
+    OS_LIBS += [
+        "m"
+    ]
+
 Library("file_recorder_gn")
