--- a/configure.in
+++ b/configure.in
@@ -4735,7 +4735,7 @@ dnl Detect yasm
 dnl ======================
 
 AC_MSG_CHECKING([for YASM assembler])
-AC_CHECK_PROGS(YASM, yasm, "")
+AC_CHECK_PROGS(YASM, yasm-1, "")
 
 if test -n "$YASM"; then
   dnl Pull out yasm's version string
@@ -4743,7 +4743,7 @@ if test -n "$YASM"; then
   _YASM_VER_FILTER='s|.* \([0-9]\{1,\}\.[0-9]\{1,\}\.[0-9]\{1,\}\.[0-9]\{1,\}\).*|\1|p'
   changequote([,])
 
-  YASM_VERSION=`yasm --version | sed -ne "$_YASM_VER_FILTER"`
+  YASM_VERSION=`yasm-1 --version | sed -ne "$_YASM_VER_FILTER"`
   _YASM_MAJOR_VERSION=`echo ${YASM_VERSION} | $AWK -F\. '{ print $1 }'`
   _YASM_MINOR_VERSION=`echo ${YASM_VERSION} | $AWK -F\. '{ print $2 }'`
   _YASM_RELEASE=`      echo ${YASM_VERSION} | $AWK -F\. '{ print $3 }'`
@@ -6425,7 +6425,7 @@ dnl --disable-libjpeg-turbo wasn't passe
 dnl doesn't exist or we have too old of a version.
 if test -n "$LIBJPEG_TURBO_X86_ASM" -o -n "$LIBJPEG_TURBO_X64_ASM" ; then
     AC_MSG_CHECKING([for YASM assembler])
-    AC_CHECK_PROGS(LIBJPEG_TURBO_AS, yasm, "")
+    AC_CHECK_PROGS(LIBJPEG_TURBO_AS, yasm-1, "")
 
     if test -z "$LIBJPEG_TURBO_AS" ; then
         AC_MSG_ERROR([yasm is required to build with libjpeg-turbo's optimized JPEG decoding routines, but you do not appear to have yasm installed.  Either install it or configure with --disable-libjpeg-turbo to use the pure C JPEG decoder.  See https://developer.mozilla.org/en/YASM for more details.])
