--- a/js/src/xpconnect/shell/xpcshell.cpp
+++ b/js/src/xpconnect/shell/xpcshell.cpp
@@ -475,8 +475,8 @@ Load(JSContext *cx, uintN argc, jsval *v
             return false;
         FILE *file = fopen(filename.ptr(), "r");
         if (!file) {
-            JS_ReportError(cx, "cannot open file '%s' for reading",
-                           filename.ptr());
+            JS_ReportError(cx, "cannot open file '%s' for reading: %s",
+                           filename.ptr(), strerror(errno));
             return false;
         }
         JSObject *scriptObj = JS_CompileFileHandleForPrincipals(cx, obj, filename.ptr(),
