Index: firefox-trunk-21.0~a1~hg20130123r119624/uriloader/exthandler/tests/unit/xpcshell.ini
===================================================================
--- firefox-trunk-21.0~a1~hg20130123r119624.orig/uriloader/exthandler/tests/unit/xpcshell.ini	2013-01-24 02:18:03.000000000 +0000
+++ firefox-trunk-21.0~a1~hg20130123r119624/uriloader/exthandler/tests/unit/xpcshell.ini	2013-01-24 20:50:54.183647945 +0000
@@ -5,6 +5,5 @@
 [test_getTypeFromExtension_ext_to_type_mapping.js]
 [test_getTypeFromExtension_with_empty_Content_Type.js]
 [test_handlerService.js]
-# Bug 676997: test consistently fails on Android
-fail-if = os == "android"
+fail-if = true
 [test_punycodeURIs.js]
