Index: firefox-trunk-22.0~a1~hg20130321r125760/uriloader/exthandler/tests/unit/xpcshell.ini
===================================================================
--- firefox-trunk-22.0~a1~hg20130321r125760.orig/uriloader/exthandler/tests/unit/xpcshell.ini	2013-03-22 18:58:09.269453730 +0000
+++ firefox-trunk-22.0~a1~hg20130321r125760/uriloader/exthandler/tests/unit/xpcshell.ini	2013-03-22 18:58:09.261453730 +0000
@@ -5,6 +5,6 @@
 [test_getTypeFromExtension_ext_to_type_mapping.js]
 [test_getTypeFromExtension_with_empty_Content_Type.js]
 [test_handlerService.js]
-# Bug 676997: test consistently fails on Android
-fail-if = os == "android"
+# https://launchpad.net/bugs/1158899
+fail-if = true
 [test_punycodeURIs.js]
Index: firefox-trunk-22.0~a1~hg20130321r125760/toolkit/components/telemetry/tests/unit/xpcshell.ini
===================================================================
--- firefox-trunk-22.0~a1~hg20130321r125760.orig/toolkit/components/telemetry/tests/unit/xpcshell.ini	2013-03-22 18:58:09.269453730 +0000
+++ firefox-trunk-22.0~a1~hg20130321r125760/toolkit/components/telemetry/tests/unit/xpcshell.ini	2013-03-22 18:58:09.261453730 +0000
@@ -3,9 +3,12 @@
 tail = 
 
 [test_nsITelemetry.js]
+skip-if = true
 [test_TelemetryPing.js]
-# Bug 676989: test fails consistently on Android
-# fail-if = os == "android"
+skip-if = true
 [test_TelemetryPing_idle.js]
+skip-if = true
 [test_TelemetryStopwatch.js]
+skip-if = true
 [test_TelemetryPingBuildID.js]
+skip-if = true
Index: firefox-trunk-22.0~a1~hg20130321r125760/toolkit/components/search/tests/xpcshell/xpcshell.ini
===================================================================
--- firefox-trunk-22.0~a1~hg20130321r125760.orig/toolkit/components/search/tests/xpcshell/xpcshell.ini	2013-03-22 02:20:12.000000000 +0000
+++ firefox-trunk-22.0~a1~hg20130321r125760/toolkit/components/search/tests/xpcshell/xpcshell.ini	2013-03-22 18:58:51.693452249 +0000
@@ -4,6 +4,8 @@
 firefox-appdir = browser
 
 [test_nocache.js]
+# https://bugzilla.mozilla.org/show_bug.cgi?id=828223#c14
+skip-if = true
 [test_645970.js]
 # Bug 845190: Too many intermittent assertions on Linux (ASSERTION: thread pool wasn't shutdown)
 skip-if = debug && os == "linux"
