Index: firefox-trunk-20.0~a1~hg20121130r114518/toolkit/mozapps/extensions/test/xpcshell/xpcshell.ini
===================================================================
--- firefox-trunk-20.0~a1~hg20121130r114518.orig/toolkit/mozapps/extensions/test/xpcshell/xpcshell.ini	2012-11-30 12:48:52.757412913 +0000
+++ firefox-trunk-20.0~a1~hg20121130r114518/toolkit/mozapps/extensions/test/xpcshell/xpcshell.ini	2012-11-30 13:41:15.617542019 +0000
@@ -231,7 +231,9 @@
 [test_update_compatmode.js]
 [test_upgrade.js]
 # Bug 676992: test consistently hangs on Android
-skip-if = os == "android"
+# This needs to write to the application directory
+skip-if = appDirReadonly || os == "android"
 [test_upgrade_strictcompat.js]
 # Bug 676992: test consistently hangs on Android
-skip-if = os == "android"
+# This needs to write to the application directory
+skip-if = appDirReadonly || os == "android"
