Index: firefox-trunk-22.0~a1~hg20130321r125760/uriloader/exthandler/tests/unit/xpcshell.ini
===================================================================
--- firefox-trunk-22.0~a1~hg20130321r125760.orig/uriloader/exthandler/tests/unit/xpcshell.ini	2013-03-22 17:41:16.441614770 +0000
+++ firefox-trunk-22.0~a1~hg20130321r125760/uriloader/exthandler/tests/unit/xpcshell.ini	2013-03-22 17:41:46.453613722 +0000
@@ -5,6 +5,6 @@
 [test_getTypeFromExtension_ext_to_type_mapping.js]
 [test_getTypeFromExtension_with_empty_Content_Type.js]
 [test_handlerService.js]
-# Bug 676997: test consistently fails on Android
-fail-if = os == "android"
+# https://launchpad.net/bugs/1158899
+fail-if = true
 [test_punycodeURIs.js]
Index: firefox-trunk-22.0~a1~hg20130321r125760/toolkit/components/telemetry/tests/unit/xpcshell.ini
===================================================================
--- firefox-trunk-22.0~a1~hg20130321r125760.orig/toolkit/components/telemetry/tests/unit/xpcshell.ini	2013-03-22 17:41:16.441614770 +0000
+++ firefox-trunk-22.0~a1~hg20130321r125760/toolkit/components/telemetry/tests/unit/xpcshell.ini	2013-03-22 17:41:16.437614770 +0000
@@ -3,9 +3,12 @@
 tail = 
 
 [test_nsITelemetry.js]
+skip-if = true
 [test_TelemetryPing.js]
-# Bug 676989: test fails consistently on Android
-# fail-if = os == "android"
+skip-if = true
 [test_TelemetryPing_idle.js]
+skip-if = true
 [test_TelemetryStopwatch.js]
+skip-if = true
 [test_TelemetryPingBuildID.js]
+skip-if = true
