Index: firefox-trunk-20.0~a1~hg20130101r117292/dom/tests/unit/xpcshell.ini
===================================================================
--- firefox-trunk-20.0~a1~hg20130101r117292.orig/dom/tests/unit/xpcshell.ini	2013-01-02 02:06:46.000000000 +0000
+++ firefox-trunk-20.0~a1~hg20130101r117292/dom/tests/unit/xpcshell.ini	2013-01-02 09:34:59.882771962 +0000
@@ -5,6 +5,8 @@
 [test_bug319968.js]
 [test_bug465752.js]
 [test_geolocation_provider.js]
+# Seems to fail intermittently on Ubuntu
+skip-if = true
 [test_geolocation_timeout.js]
 [test_geolocation_timeout_wrap.js]
 skip-if = os == "mac"
